// @storefront-ui/vue/dist/components/SfDropdown/SfDropdown.vue.mjs
import { defineComponent as g, toRefs as C, openBlock as a, createElementBlock as s, mergeProps as h, unref as e, renderSlot as l, normalizeStyle as v, normalizeClass as R, createCommentVNode as S } from "vue";
import { useTwMergeRoot as k, twMerge as V } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { useDropdown as _ } from "../../composables/useDropdown/useDropdown.mjs";
import { ClassProp as B } from "../../shared/props.mjs";
const $ = ["aria-hidden"], E = /* @__PURE__ */ g({
  inheritAttrs: !1,
  __name: "SfDropdown",
  props: {
    modelValue: {
      type: Boolean,
      default: !1
    },
    placement: {
      type: String,
      default: void 0
    },
    middleware: {
      type: Array,
      default: void 0
    },
    strategy: {
      type: String,
      default: void 0
    },
    wrapperClass: B
  },
  emits: ["update:modelValue"],
  setup(r, { emit: d }) {
    const n = r, { attrsWithoutClass: i } = k(), f = d, { modelValue: t, placement: p, middleware: m, strategy: c } = C(n), {
      referenceRef: u,
      floatingRef: y,
      style: w
    } = _({
      isOpen: t,
      placement: p,
      middleware: m,
      strategy: c,
      onClose: () => f("update:modelValue", !1)
    });
    return (o, z) => (a(), s("div", h({
      ref_key: "referenceRef",
      ref: u,
      class: e(V)("w-max", o.$attrs.class),
      "data-testid": "dropdown"
    }, e(i)), [
      l(o.$slots, "trigger"),
      e(t) ? (a(), s("div", {
        key: 0,
        ref_key: "floatingRef",
        ref: y,
        style: v(e(w)),
        "aria-hidden": !e(t) || void 0,
        "data-testid": "dropdown-content",
        class: R(r.wrapperClass)
      }, [
        l(o.$slots, "default")
      ], 14, $)) : S("", !0)
    ], 16));
  }
});
export {
  E as default
};
