// @storefront-ui/nuxt/dist/module.mjs
import { defineNuxtModule, addTemplate, installModule, addComponent, addImportsSources } from '@nuxt/kit';
import * as storefrontUi from '@storefront-ui/vue';
import { defu } from 'defu';
import { join } from 'node:path';
import '@nuxtjs/tailwindcss';

const module = defineNuxtModule({
  meta: {
    name: "storefront-ui",
    configKey: "storefrontUi",
    compatibility: {
      nuxt: ">=3.0.0"
    }
  },
  // Default configuration options of the Nuxt module
  defaults: {
    contentPath: "./node_modules/@storefront-ui/vue/**/*.{js,mjs}"
  },
  async setup(options, nuxt) {
    const { contentPath } = options;
    const customTailwindConfigTemplate = addTemplate({
      filename: "storefront-ui.tailwind.css",
      write: true,
      getContents: () => `@import "tailwindcss";
@import "@storefront-ui/vue/tailwind-config";
@plugin "@storefront-ui/typography";

@source '../**/*.vue';
@source '../node_modules/@storefront-ui/vue';
`
    });
    await installModule(
      "@nuxtjs/tailwindcss",
      defu(
        {
          configPath: [customTailwindConfigTemplate.dst, join(nuxt.options.rootDir, "tailwind.config")],
          config: { content: [contentPath ?? ""] }
        },
        nuxt.options.tailwindcss
      )
    );
    const components = [];
    const composables = [];
    Object.keys(storefrontUi).forEach((key) => {
      if (key.startsWith("Sf") && (storefrontUi[key].__name || storefrontUi[key].name)) {
        components.push(key);
      } else if (key.startsWith("use")) {
        composables.push(key);
      }
    });
    components.forEach((key) => {
      addComponent({
        name: key,
        // name of the component to be used in vue templates,
        export: key,
        filePath: `@storefront-ui/vue`
      });
    });
    addImportsSources({
      imports: composables,
      from: "@storefront-ui/vue"
    });
  }
});

export { module as default };
