// @grapesjs/studio-sdk-plugins/dist/rteTinyMce/index.es.js
var v = /* @__PURE__ */ ((t) => (t.free = "free", t.startup = "startup", t.business = "business", t.enterprise = "enterprise", t))(v || {}), m = /* @__PURE__ */ ((t) => (t.toastAdd = "studio:toastAdd", t.toastRemove = "studio:toastRemove", t.dialogOpen = "studio:dialogOpen", t.dialogClose = "studio:dialogClose", t.dialogExportCode = "studio:dialogExportCode", t.sidebarLeftSet = "studio:sidebarLeft:set", t.sidebarLeftGet = "studio:sidebarLeft:get", t.sidebarLeftToggle = "studio:sidebarLeft:toggle", t.sidebarRightSet = "studio:sidebarRight:set", t.sidebarRightGet = "studio:sidebarRight:get", t.sidebarRightToggle = "studio:sidebarRight:toggle", t.sidebarTopSet = "studio:sidebarTop:set", t.sidebarTopGet = "studio:sidebarTop:get", t.sidebarTopToggle = "studio:sidebarTop:toggle", t.sidebarBottomSet = "studio:sidebarBottom:set", t.sidebarBottomGet = "studio:sidebarBottom:get", t.sidebarBottomToggle = "studio:sidebarBottom:toggle", t.symbolAdd = "studio:symbolAdd", t.symbolDetach = "studio:symbolDetach", t.symbolOverride = "studio:symbolOverride", t.symbolPropagateStyles = "studio:propagateStyles", t.getPagesConfig = "studio:getPagesConfig", t.setPagesConfig = "studio:setPagesConfig", t.getPageSettings = "studio:getPageSettings", t.setPageSettings = "studio:setPageSettings", t.clearPage = "studio:clearPage", t.projectFiles = "studio:projectFiles", t.validateCode = "studio:validateCode", t.canvasReload = "studio:canvasReload", t.getBlocksPanel = "studio:getBlocksPanel", t.setBlocksPanel = "studio:setBlocksPanel", t.getStateContextMenu = "studio:getStateContextMenu", t.setStateContextMenu = "studio:setStateContextMenu", t.contextMenuComponent = "studio:contextMenuComponent", t.layoutAdd = "studio:layoutAdd", t.layoutRemove = "studio:layoutRemove", t.layoutToggle = "studio:layoutToggle", t.layoutUpdate = "studio:layoutUpdate", t.layoutGet = "studio:layoutGet", t.layoutConfigGet = "studio:layoutConfigGet", t.layoutConfigSet = "studio:layoutConfigSet", t.getStateTheme = "studio:getStateTheme", t.setStateTheme = "studio:setStateTheme", t.assetProviderGet = "studio:assetProviderGet", t.assetProviderAdd = "studio:assetProviderAdd", t.assetProviderRemove = "studio:assetProviderRemove", t.fontGet = "studio:fontGet", t.fontAdd = "studio:fontAdd", t.fontRemove = "studio:fontRemove", t.fontManagerOpen = "studio:fontManagerOpen", t.menuFontLoad = "studio:menuFontLoad", t.toggleStateDataSource = "studio:toggleStateDataSource", t.getStateDataSource = "studio:getStateDataSource", t.dataSourceSetGlobalData = "studio:dataSourceSetGlobalData", t.dataSourceSetImporter = "studio:dataSourceSetImporter", t.dataSourceSetExporter = "studio:dataSourceSetExporter", t.setDragAbsolute = "studio:setDragAbsolute", t))(m || {}), A = /* @__PURE__ */ ((t) => (t.dark = "dark", t.light = "light", t.auto = "auto", t))(A || {});
const L = "app.grapesjs.com", O = "app-stage.grapesjs.com", M = "app2.grapesjs.com", G = "app-stage2.grapesjs.com", R = [
  L,
  M,
  O,
  G,
  "localhost",
  "127.0.0.1",
  ".local-credentialless.webcontainer.io",
  // For stackblitz.com demos
  ".local.webcontainer.io",
  // For stackblitz.com demos
  "-sandpack.codesandbox.io"
  // For Sandpack demos
], N = "license:check:start", B = "license:check:end", U = () => typeof window < "u", j = ({ isDev: t, isStage: n }) => `${t ? "" : `https://${n ? O : L}`}/api`, W = () => {
  const t = U() && window.location.hostname;
  return !!t && (R.includes(t) || R.some((n) => t.endsWith(n)));
};
async function S({
  path: t,
  baseApiUrl: n,
  method: o = "GET",
  headers: s = {},
  params: i,
  body: l
}) {
  const c = `${n || j({ isDev: !1, isStage: !1 })}${t}`, r = {
    method: o,
    headers: {
      "Content-Type": "application/json",
      ...s
    }
  };
  l && (r.body = JSON.stringify(l));
  const b = i ? new URLSearchParams(i).toString() : "", e = b ? `?${b}` : "", a = await fetch(`${c}${e}`, r);
  if (!a.ok)
    throw new Error(`HTTP error! status: ${a.status}`);
  return a.json();
}
const P = {
  [v.free]: 0,
  [v.startup]: 10,
  [v.business]: 20,
  [v.enterprise]: 30
};
function F(t) {
  const n = t;
  return n.init = (o) => (s) => t(s, o), n;
}
const $ = (t) => /* @__PURE__ */ F(t);
async function C({
  editor: t,
  plan: n,
  pluginName: o,
  licenseKey: s,
  cleanup: i
}) {
  let l = "", d = !1;
  const c = W(), r = (e) => {
    console.warn("Cleanup plugin:", o, "Reason:", e), i();
  }, b = (e = {}) => {
    var f;
    const { error: a, sdkLicense: x } = e, u = (f = e.plan) == null ? void 0 : f.category;
    if (!(x || e.license) || a)
      r(a || "Invalid license");
    else if (u) {
      const E = P[n], p = P[u];
      E > p && r({ pluginRequiredPlan: n, licensePlan: u });
    }
  };
  t.on(N, (e) => {
    l = e == null ? void 0 : e.baseApiUrl, d = !0;
  }), t.on(B, (e) => {
    b(e);
  }), setTimeout(async () => {
    if (!d) {
      if (c) return;
      if (s) {
        const e = await K({ licenseKey: s, pluginName: o, baseApiUrl: l });
        e && b(e);
      } else
        r("The `licenseKey` option not provided");
    }
  }, 2e3);
}
async function K(t) {
  const { licenseKey: n, pluginName: o, baseApiUrl: s } = t;
  try {
    return (await S({
      baseApiUrl: s,
      path: `/sdk/${n || "na"}`,
      method: "POST",
      params: {
        d: window.location.hostname,
        pn: o
      }
    })).result || {};
  } catch (i) {
    return console.error("Error during SDK license check:", i), !1;
  }
}
const q = (t, n, o) => {
  const { doc: s = document } = o, i = o.target || s.head, l = i.querySelector(`style[${t}]`) || s.createElement("style");
  l.setAttribute(t, "true"), l.innerHTML = n, !i.contains(l) && i.appendChild(l);
}, H = (t) => t.dispatchEvent(new CustomEvent("input"));
function z(t, n) {
  if (!n) return () => {
  };
  const o = t.Components.events, s = (e) => t.getSelectedAll().includes(e), i = (e) => e.isInstanceOf("text") && !e.isChildOf("text"), l = (e) => e.isChildOf("text"), d = (e) => t.getEditing() === e, c = (e, a = {}) => {
    e.trigger("focus", a.event);
  }, r = (e, a) => {
    s(e) && i(e) && !d(e) && c(e, a);
  }, b = (e, a) => {
    if (s(e) && !i(e) && l(e)) {
      const x = e.parents().find((u) => u.isInstanceOf("text"));
      x && (c(x, a), setTimeout(() => t.select(x), 0));
    }
  };
  return t.on(o.select, r), t.on(o.selectBefore, b), () => {
    t.off(o.select, r), t.off(o.selectBefore, b);
  };
}
const V = "rteTinyMce", J = v.startup, X = `
.tox .tox-toolbar,
.tox .tox-toolbar__overflow,
.tox .tox-toolbar__primary,
.tox .tox-menu,
.tox .tox-dialog,
.tox .tox-toolbar-overlord,
.tox .tox-dialog__header,
.tox .tox-dialog__footer {
    background-color: var(--gs-theme-cl-glb-bg3);
    color: var(--gs-theme-cl-glb-txt);
}

.tox .tox-collection__item,
.tox .tox-label,
.tox .tox-toolbar-label {
    color: var(--gs-theme-cl-glb-txt);
}

.tox.tox-tinymce-inline .tox-editor-header {
    border-color: var(--gs-theme-cl-glb-br);
    background-color: var(--gs-theme-cl-glb-bg3);
}

.tox .tox-button {
    background-color: var(--gs-theme-cl-prm-bg1);
    border-color: var(--gs-theme-cl-prm-bg1);
}

.tox .tox-button:hover:not(:disabled) {
    background-color: var(--gs-theme-cl-prm-bgh);
    border-color: var(--gs-theme-cl-prm-bgh);
}

.tox .tox-button--secondary {
    background-color: var(--gs-theme-cl-glb-bg2);
    border-color: var(--gs-theme-cl-glb-br);
}

.tox .tox-button--secondary:hover:not(:disabled) {
    background-color: var(--gs-theme-cl-glb-bgh);
    border-color: var(--gs-theme-cl-glb-bgh);
}

.tox .tox-tbtn {
    color: currentColor;
}

.tox .tox-listboxfield .tox-listbox--select,
.tox .tox-textarea,
.tox .tox-tbtn--bespoke,
.tox .tox-textarea-wrap .tox-textarea:focus,
.tox .tox-textfield,
.tox .tox-toolbar-textfield {
    background-color: var(--gs-theme-cl-glb-bgh);
}

.tox .tox-custom-editor:focus-within,
.tox .tox-listboxfield .tox-listbox--select:focus,
.tox .tox-textarea-wrap:focus-within,
.tox .tox-textarea:focus,
.tox .tox-textfield:focus {
    background-color: var(--gs-theme-cl-glb-bgh);
    border-color: var(--gs-theme-cl-prm-bg1);
}

.tox .tox-tbtn:hover,
.tox .tox-tbtn--enabled,
.tox .tox-tbtn--enabled:hover,
.tox .tox-tbtn:active,
.tox .tox-tbtn:focus,
.tox .tox-swatches__picker-btn:hover,
.tox .tox-collection--list .tox-collection__item--enabled,
.tox .tox-collection--toolbar .tox-collection__item--enabled,
.tox .tox-split-button:focus {
    background-color: var(--gs-theme-cl-prm-bg1);
    color: var(--gs-theme-cl-prm-txt);
}

.tox .tox-collection--list .tox-collection__item--active,
.tox .tox-collection--toolbar .tox-collection__item--active {
    background-color: var(--gs-theme-cl-prm-bgh);
    color: var(--gs-theme-cl-prm-txt) !important;
}

.tox .tox-split-button:hover {
    box-shadow: 0 0 0 1px var(--gs-theme-cl-glb-bgh) inset;
}

.tox .tox-dialog__body-nav-item--active {
    color: var(--gs-theme-cl-prm-bg1);
    border-color: var(--gs-theme-cl-prm-bg1);
}

.tox .tox-tbtn svg,
.tox .tox-tbtn:hover svg {
    fill: currentColor;
}

.tox .tox-insert-table-picker .tox-insert-table-picker__selected {
    background-color: var(--gs-theme-cl-prm-bg1);
    border-color: var(--gs-theme-cl-prm-bg3);
    opacity: 0.75;
}`, T = /* @__PURE__ */ new WeakMap(), Y = (t, n) => {
  T.set(t, n);
}, Q = (t) => {
  T.delete(t);
}, w = (t) => T.get(t), y = (t) => {
  const n = t.target, o = n.closest(".tox"), s = n.matches(".ephox-snooker-resizer-bar");
  (o || s) && t.stopPropagation();
}, _ = (t) => {
  t.stopPropagation(), t.preventDefault();
}, Z = (t, n) => {
  const o = t.getDoc(), s = t.getWin();
  if (n != null && n.event) {
    const { event: i } = n, { clientX: l, clientY: d } = i;
    let c;
    if (o.caretRangeFromPoint)
      c = o.caretRangeFromPoint(l, d);
    else if (o.caretPositionFromPoint) {
      const r = o.caretPositionFromPoint(l, d);
      c = o.createRange(), c == null || c.setStart(r.offsetNode, r.offset);
    }
    if (c) {
      const r = s.getSelection();
      r == null || r.removeAllRanges(), r == null || r.addRange(c), t.selection.setRng(c);
    }
  } else
    t.execCommand("SelectAll");
}, tt = (t) => {
  var b, e;
  const { el: n, editor: o, componentView: s } = t, i = o.runCommand(m.getStateTheme), l = s.model, d = {
    target: n,
    inline: !0,
    skin: (i == null ? void 0 : i.theme) === A.dark ? "oxide-dark" : "oxide",
    menubar: !1,
    image_advtab: !0,
    paste_data_images: !0,
    statusbar: !1,
    branding: !1,
    // It's possible to place it inside the GJS container, but it has
    // issues with absolute elements like select/colorPicker with scrolled iframe
    // fixed_toolbar_container_target: editor.RichTextEditor.toolbar,
    forced_root_block: "div",
    // avoid default <p> tag around textnodes
    plugins: "advlist autolink lists link image media table",
    toolbar_mode: "sliding",
    toolbar: "bold italic underline strikethrough forecolor backcolor alignleft aligncenter alignright alignjustify link image | fontfamily fontsize bullist numlist outdent indent table"
  }, c = ((e = (b = t.pluginOpts) == null ? void 0 : b.loadConfig) == null ? void 0 : e.call(b, { config: d, editor: o, componentView: s, component: l })) || {};
  return {
    ...d,
    ...c
  };
}, D = (t, n = document) => new Promise((o, s) => {
  const i = Array.isArray(t) ? [...t] : [t], { head: l } = n, d = (c) => {
    if (c.length) {
      const r = n.createElement("script");
      r.src = c.shift(), r.onload = d.bind(null, c), r.onerror = s, l.appendChild(r);
    } else
      o();
  };
  d(i);
}), et = function(t, n = {}) {
  const o = {
    cdnScript: "https://cdn.jsdelivr.net/npm/tinymce@6.8.5/tinymce.min.js",
    ...n
  }, { cdnScript: s } = o, i = t.Canvas.events, l = !o.skipCustomTheme, d = "gs-plg-tinymce-studio-theme", c = async (e, a) => {
    const x = e.ownerDocument, u = x.defaultView, g = a == null ? void 0 : a.view;
    if (!u || !g) {
      console.warn("Missing required data", { win: u, componentView: g });
      return;
    }
    const f = tt({
      el: e,
      editor: t,
      componentView: g,
      pluginOpts: o
    }), p = (await u.tinymce.init(f))[0];
    if (!p) return;
    const I = (await window.tinymce.init({
      ...f,
      target: document.createElement("div")
    }))[0];
    return p.ui.show(), x.body.addEventListener("mousedown", y), document.body.addEventListener("mousedown", y), p.on("Change", () => H(e)), p.on("blur", () => p.ui.show()), p.on("OpenWindow", function() {
      var k;
      const h = (k = x.querySelector(".tox-dialog")) == null ? void 0 : k.closest(".tox");
      h && (h == null ? void 0 : h.ownerDocument) !== document && (document.body.appendChild(h), p.once("CloseWindow", () => {
        h && h.ownerDocument === document && x.body.appendChild(h);
      }));
    }), e.addEventListener("dragstart", _, { capture: !0 }), t.runCommand(m.setStateContextMenu, { enabled: !1 }), Z(p, a), { rteEditor: p, rteEditorShallow: I };
  }, r = (e, a) => {
    e.ownerDocument.body.removeEventListener("mousedown", y), document.body.removeEventListener("mousedown", y), e.removeEventListener("dragstart", _, { capture: !0 }), t.runCommand(m.setStateContextMenu, { enabled: !0 }), a.rteEditor.destroy(), a.rteEditorShallow.destroy();
  };
  t.on(i.frameLoad, async ({ window: e }) => {
    const a = e.document;
    D(s, a);
  }), t.onReady(() => {
    D(s, document), l && q(d, X, { target: document.body });
  }), t.setCustomRte({
    parseContent: !0,
    // @ts-expect-error - Type 'CustomRteOptions' is missing the following properties from type 'CustomRteOptions': editor, component, el
    async enable(e, a, x) {
      const u = w(e);
      if (u) return u.rteEditor;
      const g = await c(e, x);
      return g && Y(e, g), g == null ? void 0 : g.rteEditor;
    },
    disable(e) {
      const a = w(e);
      a && (Q(e), r(e, a));
    },
    getContent(e, a, x) {
      var f;
      const u = w(e), g = u == null ? void 0 : u.rteEditor;
      return g && !g.isDirty() ? x.view.lastContent : g ? (f = g.getContent) == null ? void 0 : f.call(g) : e.innerHTML;
    }
  });
  const b = z(t, o.enableOnClick);
  C({
    editor: t,
    licenseKey: o.licenseKey,
    plan: J,
    pluginName: V,
    cleanup: () => {
      t.RichTextEditor.customRte = void 0, b();
    }
  });
}, rt = $(et);
export {
  rt as default
};
