// @polar-sh/nuxt/dist/runtime/server/checkoutHandler.js
import { Polar } from "@polar-sh/sdk";
import { createError, getValidatedQuery, sendRedirect } from "h3";
import { z } from "zod";
const checkoutQuerySchema = z.object({
  products: z.string().transform((value) => value.split(",")).pipe(z.string().array()),
  customerId: z.string().nonempty().optional(),
  customerExternalId: z.string().nonempty().optional(),
  customerEmail: z.string().email().optional(),
  customerName: z.string().nonempty().optional(),
  customerBillingAddress: z.string().nonempty().optional(),
  customerTaxId: z.string().nonempty().optional(),
  customerIpAddress: z.string().nonempty().optional(),
  customerMetadata: z.string().nonempty().optional(),
  allowDiscountCodes: z.string().toLowerCase().transform((x) => x === "true").pipe(z.boolean()).optional(),
  discountId: z.string().nonempty().optional(),
  metadata: z.string().nonempty().optional()
});
export const Checkout = ({
  accessToken,
  successUrl,
  returnUrl,
  server,
  theme,
  includeCheckoutId = true
}) => {
  return async (event) => {
    const {
      products,
      customerId,
      customerExternalId,
      customerEmail,
      customerName,
      customerBillingAddress,
      customerTaxId,
      customerIpAddress,
      customerMetadata,
      allowDiscountCodes,
      discountId,
      metadata
    } = await getValidatedQuery(event, checkoutQuerySchema.parse);
    try {
      const success = successUrl ? new URL(successUrl) : void 0;
      if (success && includeCheckoutId) {
        success.searchParams.set("checkoutId", "{CHECKOUT_ID}");
      }
      const retUrl = returnUrl ? new URL(returnUrl) : void 0;
      const polar = new Polar({ accessToken, server });
      const result = await polar.checkouts.create({
        products,
        successUrl: success ? decodeURI(success.toString()) : void 0,
        customerId,
        externalCustomerId: customerExternalId,
        customerEmail,
        customerName,
        customerBillingAddress: customerBillingAddress ? JSON.parse(customerBillingAddress) : void 0,
        customerTaxId,
        customerIpAddress,
        customerMetadata: customerMetadata ? JSON.parse(customerMetadata) : void 0,
        allowDiscountCodes,
        discountId,
        metadata: metadata ? JSON.parse(metadata) : void 0,
        returnUrl: retUrl ? decodeURI(retUrl.toString()) : void 0
      });
      const redirectUrl = new URL(result.url);
      if (theme) {
        redirectUrl.searchParams.set("theme", theme);
      }
      return sendRedirect(event, redirectUrl.toString());
    } catch (error) {
      console.error("Failed to checkout:", error);
      throw createError({
        statusCode: 500,
        statusMessage: error.message,
        message: error.message ?? "Internal server error"
      });
    }
  };
};
