// @grapesjs/studio-sdk-plugins/dist/youtubeAssetProvider/index.es.js
var g = /* @__PURE__ */ ((e) => (e.free = "free", e.startup = "startup", e.business = "business", e.enterprise = "enterprise", e))(g || {}), h = /* @__PURE__ */ ((e) => (e.toastAdd = "studio:toastAdd", e.toastRemove = "studio:toastRemove", e.dialogOpen = "studio:dialogOpen", e.dialogClose = "studio:dialogClose", e.dialogExportCode = "studio:dialogExportCode", e.sidebarLeftSet = "studio:sidebarLeft:set", e.sidebarLeftGet = "studio:sidebarLeft:get", e.sidebarLeftToggle = "studio:sidebarLeft:toggle", e.sidebarRightSet = "studio:sidebarRight:set", e.sidebarRightGet = "studio:sidebarRight:get", e.sidebarRightToggle = "studio:sidebarRight:toggle", e.sidebarTopSet = "studio:sidebarTop:set", e.sidebarTopGet = "studio:sidebarTop:get", e.sidebarTopToggle = "studio:sidebarTop:toggle", e.sidebarBottomSet = "studio:sidebarBottom:set", e.sidebarBottomGet = "studio:sidebarBottom:get", e.sidebarBottomToggle = "studio:sidebarBottom:toggle", e.symbolAdd = "studio:symbolAdd", e.symbolDetach = "studio:symbolDetach", e.symbolOverride = "studio:symbolOverride", e.symbolPropagateStyles = "studio:propagateStyles", e.getPagesConfig = "studio:getPagesConfig", e.setPagesConfig = "studio:setPagesConfig", e.getPageSettings = "studio:getPageSettings", e.setPageSettings = "studio:setPageSettings", e.clearPage = "studio:clearPage", e.projectFiles = "studio:projectFiles", e.validateCode = "studio:validateCode", e.canvasReload = "studio:canvasReload", e.getBlocksPanel = "studio:getBlocksPanel", e.setBlocksPanel = "studio:setBlocksPanel", e.getStateContextMenu = "studio:getStateContextMenu", e.setStateContextMenu = "studio:setStateContextMenu", e.contextMenuComponent = "studio:contextMenuComponent", e.layoutAdd = "studio:layoutAdd", e.layoutRemove = "studio:layoutRemove", e.layoutToggle = "studio:layoutToggle", e.layoutUpdate = "studio:layoutUpdate", e.layoutGet = "studio:layoutGet", e.layoutConfigGet = "studio:layoutConfigGet", e.layoutConfigSet = "studio:layoutConfigSet", e.getStateTheme = "studio:getStateTheme", e.setStateTheme = "studio:setStateTheme", e.assetProviderGet = "studio:assetProviderGet", e.assetProviderAdd = "studio:assetProviderAdd", e.assetProviderRemove = "studio:assetProviderRemove", e.fontGet = "studio:fontGet", e.fontAdd = "studio:fontAdd", e.fontRemove = "studio:fontRemove", e.fontManagerOpen = "studio:fontManagerOpen", e.menuFontLoad = "studio:menuFontLoad", e.toggleStateDataSource = "studio:toggleStateDataSource", e.getStateDataSource = "studio:getStateDataSource", e.dataSourceSetGlobalData = "studio:dataSourceSetGlobalData", e.dataSourceSetImporter = "studio:dataSourceSetImporter", e.dataSourceSetExporter = "studio:dataSourceSetExporter", e.setDragAbsolute = "studio:setDragAbsolute", e))(h || {});
const T = "app.grapesjs.com", A = "app-stage.grapesjs.com", I = "app2.grapesjs.com", k = "app-stage2.grapesjs.com", f = [
  T,
  I,
  A,
  k,
  "localhost",
  "127.0.0.1",
  ".local-credentialless.webcontainer.io",
  // For stackblitz.com demos
  ".local.webcontainer.io",
  // For stackblitz.com demos
  "-sandpack.codesandbox.io"
  // For Sandpack demos
], x = "license:check:start", U = "license:check:end", D = () => typeof window < "u", G = ({ isDev: e, isStage: o }) => `${e ? "" : `https://${o ? A : T}`}/api`, O = () => {
  const e = D() && window.location.hostname;
  return !!e && (f.includes(e) || f.some((o) => e.endsWith(o)));
};
async function E({
  path: e,
  baseApiUrl: o,
  method: a = "GET",
  headers: i = {},
  params: c,
  body: u
}) {
  const t = `${o || G({ isDev: !1, isStage: !1 })}${e}`, n = {
    method: a,
    headers: {
      "Content-Type": "application/json",
      ...i
    }
  };
  u && (n.body = JSON.stringify(u));
  const l = c ? new URLSearchParams(c).toString() : "", s = l ? `?${l}` : "", r = await fetch(`${t}${s}`, n);
  if (!r.ok)
    throw new Error(`HTTP error! status: ${r.status}`);
  return r.json();
}
const v = {
  [g.free]: 0,
  [g.startup]: 10,
  [g.business]: 20,
  [g.enterprise]: 30
};
function M(e) {
  const o = e;
  return o.init = (a) => (i) => e(i, a), o;
}
const $ = (e) => /* @__PURE__ */ M(e);
async function j({
  editor: e,
  plan: o,
  pluginName: a,
  licenseKey: i,
  cleanup: c
}) {
  let u = "", p = !1;
  const t = O(), n = (s) => {
    console.warn("Cleanup plugin:", a, "Reason:", s), c();
  }, l = (s = {}) => {
    var b;
    const { error: r, sdkLicense: y } = s, d = (b = s.plan) == null ? void 0 : b.category;
    if (!(y || s.license) || r)
      n(r || "Invalid license");
    else if (d) {
      const L = v[o], R = v[d];
      L > R && n({ pluginRequiredPlan: o, licensePlan: d });
    }
  };
  e.on(x, (s) => {
    u = s == null ? void 0 : s.baseApiUrl, p = !0;
  }), e.on(U, (s) => {
    l(s);
  }), setTimeout(async () => {
    if (!p) {
      if (t) return;
      if (i) {
        const s = await B({ licenseKey: i, pluginName: a, baseApiUrl: u });
        s && l(s);
      } else
        n("The `licenseKey` option not provided");
    }
  }, 2e3);
}
async function B(e) {
  const { licenseKey: o, pluginName: a, baseApiUrl: i } = e;
  try {
    return (await E({
      baseApiUrl: i,
      path: `/sdk/${o || "na"}`,
      method: "POST",
      params: {
        d: window.location.hostname,
        pn: a
      }
    })).result || {};
  } catch (c) {
    return console.error("Error during SDK license check:", c), !1;
  }
}
const N = (e) => {
  const o = {};
  for (const a of Object.keys(e)) {
    let i = e[a];
    i && (Array.isArray(i) ? i = i.map((c) => c.toString()).join(",") : i = i.toString(), o[a] = i);
  }
  return new URLSearchParams(o);
}, P = "youtube-video", W = (e) => {
  const { apiKey: o, searchParams: a, thumbnailQuality: i = "high" } = e, c = (t = {}) => {
    const l = "https://www.googleapis.com/youtube/v3/search", s = N(t);
    return `${l}?${s}`;
  }, u = (t) => {
    var n, l, s;
    return ((n = t[i]) == null ? void 0 : n.url) || ((l = t.high) == null ? void 0 : l.url) || ((s = t.medium) == null ? void 0 : s.url) || t.default.url;
  }, p = (t) => {
    const n = u(t.snippet.thumbnails);
    return {
      id: t.id.videoId,
      src: `https://www.youtube.com/watch?v=${t.id.videoId}`,
      name: t.snippet.title,
      type: P,
      customData: { thumbnailUrl: n, videoItem: t }
    };
  };
  return {
    id: "youtube",
    types: [P],
    label: ({ editor: t }) => t.I18n.t("youtubeAssetProvider.providerLabel"),
    search: { reloadOnInput: !0, debounceMs: 1e3 },
    async onLoad({ searchValue: t, pageCustomData: n }) {
      var b;
      const l = {
        maxResults: 25,
        q: t,
        pageToken: n == null ? void 0 : n.token,
        type: ["video"],
        key: o
      }, s = a == null ? void 0 : a({ searchValue: t }), r = {
        ...l,
        ...s
      };
      r.part ? r.part.includes("snippet") || r.part.push("snippet") : r.part = ["snippet"];
      const d = await (await fetch(c(r))).json();
      return (b = d.items) != null && b.length ? {
        items: d.items.map(p),
        nextPageCustomData: { token: d.nextPageToken },
        isLastPage: !d.nextPageToken
      } : { items: [], isLastPage: !0 };
    },
    itemLayout: ({ assetProps: t, onSelect: n }) => ({
      type: "column",
      onClick: () => n(t),
      htmlAttrs: {
        title: t.name
      },
      style: {
        borderWidth: "1px",
        borderRadius: 8,
        overflow: "hidden",
        cursor: "pointer"
      },
      children: [
        {
          type: "custom",
          render: () => {
            var l;
            return `<img src="${(l = t.customData) == null ? void 0 : l.thumbnailUrl}" alt="${t.name}" style="width: 100%; height: 100px; object-fit: cover">`;
          }
        },
        {
          type: "text",
          content: t.name,
          style: {
            overflow: "hidden",
            textOverflow: "ellipsis",
            whiteSpace: "nowrap",
            paddingLeft: 8,
            paddingRight: 8
          }
        }
      ]
    })
  };
}, Y = {
  youtubeAssetProvider: {
    searchBtn: "Search on YouTube",
    modalTitle: "Select YouTube Video",
    providerLabel: "YouTube Videos"
  }
}, _ = "youtubeAssetProvider", K = g.startup, F = function(e, o) {
  var l;
  const { skipVideoComponent: a, i18n: i = {} } = o, c = W(o);
  e.runCommand(h.assetProviderAdd, { provider: c });
  const u = "search-on-youtube", p = "video", t = (l = e.Components.getType(p)) == null ? void 0 : l.model.prototype, n = (s) => {
    e.Assets.open({
      modalTitle: e.I18n.t("youtubeAssetProvider.modalTitle"),
      providerId: c.id,
      typeId: c.types[0],
      providers: [],
      types: [],
      select: (r) => {
        e.Assets.close(), s.set({ videoId: r.attributes.id });
      },
      content: {
        header: {
          addUrl: !1,
          upload: !1
        }
      }
    });
  };
  !a && t && e.Components.addType(p, {
    model: {
      getYoutubeTraits() {
        const [s, ...r] = t.getYoutubeTraits.apply(this);
        return [
          s,
          {
            id: u,
            type: "button",
            label: e.I18n.t("youtubeAssetProvider.searchBtn"),
            command: () => n(this)
          },
          ...r
        ];
      }
    },
    view: {
      events: () => ({
        dblclick: "onActive"
      }),
      onActive() {
        const { model: s } = this, r = s.get("provider") || "";
        r != null && r.startsWith("yt") && n(s);
      }
    }
  }), e.I18n.addMessages({
    en: Y,
    ...i
  }), j({
    editor: e,
    licenseKey: o.licenseKey,
    plan: K,
    pluginName: _,
    cleanup: () => {
      e.runCommand(h.assetProviderRemove, { id: c.id }), !a && t && e.Components.addType(p, {
        model: {
          getYoutubeTraits() {
            return t.getYoutubeTraits.apply(this).filter((s) => s.id !== u);
          }
        }
      });
    }
  });
}, q = $(F);
export {
  q as default
};
