// @storefront-ui/vue/dist/components/SfButton/SfButton.vue.mjs
import { defineComponent as c, toRefs as g, computed as o, openBlock as h, createBlock as w, resolveDynamicComponent as x, mergeProps as C, unref as d, withCtx as B, renderSlot as l, createCommentVNode as u } from "vue";
import { useTwMergeRoot as S, twMerge as k } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { useAttrsRef as $ } from "../../shared/reactiveContext.mjs";
import { SfButtonSize as n, SfButtonVariant as e } from "@storefront-ui/shared";
const z = {
  [e.primary]: "shadow-sm hover:shadow-md active:shadow-sm disabled:bg-disabled-300",
  [e.secondary]: "ring-1 ring-inset shadow-sm hover:shadow-md active:shadow-sm disabled:ring-1 disabled:ring-disabled-300 disabled:bg-white/50",
  [e.tertiary]: "disabled:bg-transparent"
}, R = {
  [e.primary]: "text-white bg-primary-700 hover:bg-primary-800 active:bg-primary-900",
  [e.secondary]: "text-primary-700 hover:bg-primary-100 hover:text-primary-800 active:bg-primary-200 active:text-primary-900 ring-primary-700 hover:ring-primary-800 active:ring-primary-900",
  [e.tertiary]: "text-primary-700 hover:bg-primary-100 hover:text-primary-800 active:bg-primary-200 active:text-primary-900"
}, V = /* @__PURE__ */ c({
  inheritAttrs: !1,
  __name: "SfButton",
  props: {
    size: {
      type: String,
      default: n.base
    },
    variant: {
      type: String,
      default: e.primary
    },
    blank: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    square: {
      type: Boolean,
      default: !1
    },
    tag: {
      type: [String, Object],
      default: void 0
    }
  },
  setup(a) {
    const p = a, { attrsWithoutClass: m } = S(), { size: f, tag: r, square: s } = g(p), y = o(() => {
      switch (f.value) {
        case n.sm:
          return [s.value ? "p-1.5" : "leading-5 text-sm py-1.5 px-3", "gap-1.5"];
        case n.lg:
          return [s.value ? "p-3" : "py-3 leading-6 px-6", "gap-3"];
        default:
          return [s.value ? "p-2" : "py-2 leading-6 px-4", "gap-2"];
      }
    }), i = o(() => (r == null ? void 0 : r.value) || "button"), b = $(), v = o(
      () => b.value.type ?? (typeof i.value == "string" && i.value.toLowerCase() === "button" ? "button" : void 0)
    );
    return (t, j) => (h(), w(x(i.value), C({ "data-testid": "button" }, d(m), {
      type: v.value,
      disabled: a.disabled,
      class: d(k)(
        "inline-flex items-center justify-center font-medium text-base focus-visible:outline focus-visible:outline-offset rounded-full disabled:text-disabled-500 disabled:bg-disabled-300 disabled:shadow-none disabled:ring-0 disabled:cursor-not-allowed",
        ...y.value,
        z[a.variant],
        { [R[a.variant]]: !a.blank },
        t.$attrs.class
      )
    }), {
      default: B(() => [
        t.$slots.prefix ? l(t.$slots, "prefix", { key: 0 }) : u("", !0),
        l(t.$slots, "default"),
        t.$slots.suffix ? l(t.$slots, "suffix", { key: 1 }) : u("", !0)
      ]),
      _: 3
    }, 16, ["type", "disabled", "class"]));
  }
});
export {
  V as default,
  z as variantClasses,
  R as variantColorClasses
};
