// @polar-sh/sdk/dist/commonjs/models/operations/ordersexport.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrdersExportRequest$ = exports.OrdersExportRequest$outboundSchema = exports.OrdersExportRequest$inboundSchema = exports.OrdersExportQueryParamProductIDFilter$ = exports.OrdersExportQueryParamProductIDFilter$outboundSchema = exports.OrdersExportQueryParamProductIDFilter$inboundSchema = exports.OrdersExportQueryParamOrganizationIDFilter$ = exports.OrdersExportQueryParamOrganizationIDFilter$outboundSchema = exports.OrdersExportQueryParamOrganizationIDFilter$inboundSchema = void 0;
exports.ordersExportQueryParamOrganizationIDFilterToJSON = ordersExportQueryParamOrganizationIDFilterToJSON;
exports.ordersExportQueryParamOrganizationIDFilterFromJSON = ordersExportQueryParamOrganizationIDFilterFromJSON;
exports.ordersExportQueryParamProductIDFilterToJSON = ordersExportQueryParamProductIDFilterToJSON;
exports.ordersExportQueryParamProductIDFilterFromJSON = ordersExportQueryParamProductIDFilterFromJSON;
exports.ordersExportRequestToJSON = ordersExportRequestToJSON;
exports.ordersExportRequestFromJSON = ordersExportRequestFromJSON;
const z = __importStar(require("zod/v3"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.OrdersExportQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrdersExportQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersExportQueryParamOrganizationIDFilter$;
(function (OrdersExportQueryParamOrganizationIDFilter$) {
    /** @deprecated use `OrdersExportQueryParamOrganizationIDFilter$inboundSchema` instead. */
    OrdersExportQueryParamOrganizationIDFilter$.inboundSchema = exports.OrdersExportQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `OrdersExportQueryParamOrganizationIDFilter$outboundSchema` instead. */
    OrdersExportQueryParamOrganizationIDFilter$.outboundSchema = exports.OrdersExportQueryParamOrganizationIDFilter$outboundSchema;
})(OrdersExportQueryParamOrganizationIDFilter$ || (exports.OrdersExportQueryParamOrganizationIDFilter$ = OrdersExportQueryParamOrganizationIDFilter$ = {}));
function ordersExportQueryParamOrganizationIDFilterToJSON(ordersExportQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.OrdersExportQueryParamOrganizationIDFilter$outboundSchema.parse(ordersExportQueryParamOrganizationIDFilter));
}
function ordersExportQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrdersExportQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
exports.OrdersExportQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrdersExportQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersExportQueryParamProductIDFilter$;
(function (OrdersExportQueryParamProductIDFilter$) {
    /** @deprecated use `OrdersExportQueryParamProductIDFilter$inboundSchema` instead. */
    OrdersExportQueryParamProductIDFilter$.inboundSchema = exports.OrdersExportQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `OrdersExportQueryParamProductIDFilter$outboundSchema` instead. */
    OrdersExportQueryParamProductIDFilter$.outboundSchema = exports.OrdersExportQueryParamProductIDFilter$outboundSchema;
})(OrdersExportQueryParamProductIDFilter$ || (exports.OrdersExportQueryParamProductIDFilter$ = OrdersExportQueryParamProductIDFilter$ = {}));
function ordersExportQueryParamProductIDFilterToJSON(ordersExportQueryParamProductIDFilter) {
    return JSON.stringify(exports.OrdersExportQueryParamProductIDFilter$outboundSchema.parse(ordersExportQueryParamProductIDFilter));
}
function ordersExportQueryParamProductIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrdersExportQueryParamProductIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportQueryParamProductIDFilter' from JSON`);
}
/** @internal */
exports.OrdersExportRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
    });
});
/** @internal */
exports.OrdersExportRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersExportRequest$;
(function (OrdersExportRequest$) {
    /** @deprecated use `OrdersExportRequest$inboundSchema` instead. */
    OrdersExportRequest$.inboundSchema = exports.OrdersExportRequest$inboundSchema;
    /** @deprecated use `OrdersExportRequest$outboundSchema` instead. */
    OrdersExportRequest$.outboundSchema = exports.OrdersExportRequest$outboundSchema;
})(OrdersExportRequest$ || (exports.OrdersExportRequest$ = OrdersExportRequest$ = {}));
function ordersExportRequestToJSON(ordersExportRequest) {
    return JSON.stringify(exports.OrdersExportRequest$outboundSchema.parse(ordersExportRequest));
}
function ordersExportRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrdersExportRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportRequest' from JSON`);
}
