// @storefront-ui/vue/dist/components/SfLink/SfLink.vue.mjs
import { defineComponent as o, openBlock as s, createBlock as i, resolveDynamicComponent as l, mergeProps as c, unref as a, withCtx as u, renderSlot as f } from "vue";
import { useTwMergeRoot as d, twMerge as m } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfLinkVariant as e } from "@storefront-ui/shared";
const p = {
  [e.primary]: "text-secondary-700 hover:text-secondary-800 active:text-secondary-700",
  [e.secondary]: "text-neutral-700 hover:text-neutral-800 active:text-neutral-700"
}, h = /* @__PURE__ */ o({
  inheritAttrs: !1,
  __name: "SfLink",
  props: {
    tag: {
      type: [String, Object],
      default: "a"
    },
    variant: {
      type: String,
      default: e.primary
    }
  },
  setup(t) {
    const { attrsWithoutClass: n } = d();
    return (r, v) => (s(), i(l(t.tag), c({
      class: a(m)(
        "focus-visible:outline focus-visible:outline-offset focus-visible:rounded-xs underline hover:no-underline",
        p[t.variant],
        r.$attrs.class
      ),
      "data-testid": "link"
    }, a(n)), {
      default: u(() => [
        f(r.$slots, "default")
      ]),
      _: 3
    }, 16, ["class"]));
  }
});
export {
  h as default
};
