// @storefront-ui/vue/dist/composables/useTooltip/useTooltip.mjs
import { ref as R, computed as e, toValue as $, unref as D } from "vue";
import { offset as M, shift as O, flip as k, arrow as z } from "@floating-ui/vue";
import { onKeyStroke as E } from "@vueuse/core";
import { useDisclosure as K } from "../useDisclosure/useDisclosure.mjs";
import { usePopover as T } from "../usePopover/usePopover.mjs";
function A(u) {
  const {
    placement: w = "top",
    strategy: g,
    middleware: y,
    arrowSize: o = "6px",
    ...d
  } = u || {}, t = R(), { isOpen: a, open: p, close: r, toggle: v } = K(), {
    style: s,
    middlewareData: n,
    placement: b,
    referenceRef: l,
    floatingRef: i
  } = T({
    isOpen: a,
    placement: w,
    strategy: g,
    middleware: e(() => [
      ...$(y) || [M(8), O(), k()],
      z({ element: t })
    ]),
    ...d
  });
  function f() {
    if (n.value.arrow) {
      const { x: m, y: c } = n.value.arrow, S = D(b).split("-")[0];
      return {
        position: "absolute",
        width: o,
        height: o,
        top: typeof c == "number" ? `${c}px` : "",
        left: typeof m == "number" ? `${m}px` : "",
        [{
          top: "bottom",
          right: "left",
          bottom: "top",
          left: "right"
        }[S]]: `calc(${o} / -2)`
      };
    }
    return {};
  }
  const h = e(() => ({
    ref: l,
    onMouseenter: p,
    onMouseleave: r
  })), P = e(() => ({
    ref: i,
    style: s.value
  })), x = e(() => ({
    ref: t,
    style: f()
  }));
  return E("Escape", r, { target: l }), {
    referenceRef: l,
    floatingRef: i,
    arrowRef: t,
    style: e(() => ({
      floating: s.value,
      arrow: f()
    })),
    middlewareData: n,
    isOpen: a,
    open: p,
    close: r,
    toggle: v,
    triggerProps: h,
    tooltipProps: P,
    arrowProps: x
  };
}
export {
  A as useTooltip
};
