// @storefront-ui/vue/dist/composables/useTrapFocus/useTrapFocus.mjs
import { unrefElement as g } from "@vueuse/core";
import { tabbable as F } from "tabbable";
import { ref as n, watch as N } from "vue";
import { focusNext as k, focusPrev as h, isTab as C, isTabAndShift as G } from "@storefront-ui/shared";
import { waitForNextRender as R } from "../../shared/render.mjs";
var U = /* @__PURE__ */ ((r) => (r.autofocus = "autofocus", r.container = "container", r))(U || {});
const P = {
  trapTabs: !0,
  activeState: n(!0),
  initialFocus: 0,
  initialFocusContainerFallback: !1,
  arrowKeysLeftRight: !1,
  arrowKeysUpDown: !1
}, B = (r, A) => {
  const {
    trapTabs: d,
    arrowFocusGroupSelector: c,
    includeContainer: v,
    activeState: x,
    initialFocus: u,
    arrowKeysLeftRight: w,
    arrowKeysUpDown: b,
    initialFocusContainerFallback: D
  } = {
    ...P,
    ...A
  }, i = n(), s = n([]);
  let e;
  const p = () => {
    i.value = document.activeElement;
  }, f = ({
    event: o,
    additionalData: a
  }) => h({
    current: i.value,
    focusables: s.value,
    event: o,
    ...a
  }), l = ({
    event: o,
    additionalData: a
  }) => k({
    current: i.value,
    focusables: s.value,
    event: o,
    ...a
  }), y = (o) => {
    const t = c && (e == null ? void 0 : e.querySelector(c)) ? { arrowFocusGroupSelector: c } : {};
    w && o.key === "ArrowLeft" && f({ additionalData: t }), w && o.key === "ArrowRight" && l({ additionalData: t }), b && o.key === "ArrowUp" && f({ additionalData: t }), b && o.key === "ArrowDown" && l({ additionalData: t }), d && C(o) && l({ event: o }), d && G(o) && f({ event: o });
  }, K = () => {
    e == null || e.removeEventListener("keydown", y), e == null || e.removeEventListener("focus", p, !0);
  };
  return N(
    [r, x],
    async ([o, a]) => {
      if (o && a) {
        let t = !1;
        if (await R(), e = g(o), e == null || e.addEventListener("focus", p, !0), e == null || e.addEventListener("keydown", y), s.value = F(e, { includeContainer: v }), typeof u == "number")
          s.value[u] ? s.value[u].focus() : (console.error(`There is no element with given index ${u}`), t = !0);
        else if (u === "autofocus") {
          const m = s.value.find((S) => S.hasAttribute("autofocus"));
          m ? m.focus() : t = !0;
        }
        (D && t || u === "container") && (e == null || e.focus());
      } else
        s.value = [], i.value = void 0, K();
    },
    { immediate: !0 }
  ), {
    current: i,
    focusables: s,
    focusNext: k,
    focusPrev: h,
    updateFocusableElements: () => {
      s.value = F(r.value, { includeContainer: v });
    }
  };
};
export {
  U as InitialFocusType,
  B as useTrapFocus
};
