// @storefront-ui/vue/dist/components/SfRadio/SfRadio.vue.mjs
import { defineComponent as u, toRefs as b, computed as m, withDirectives as v, openBlock as p, createElementBlock as f, mergeProps as g, unref as d, vModelRadio as h } from "vue";
import { useTwMergeRoot as y, twMerge as k } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
const w = ["name", "value", "disabled"], R = /* @__PURE__ */ u({
  inheritAttrs: !1,
  __name: "SfRadio",
  props: {
    modelValue: {
      type: String,
      default: ""
    },
    name: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    value: {
      type: String,
      default: ""
    },
    invalid: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: o }) {
    const i = e, l = o, { attrsWithoutClass: n } = y(), { modelValue: c } = b(i), r = m({
      get: () => c.value,
      set: (a) => {
        l("update:modelValue", a);
      }
    });
    return (a, t) => v((p(), f("input", g({
      "onUpdate:modelValue": t[0] || (t[0] = (s) => r.value = s),
      name: e.name,
      type: "radio",
      value: e.value,
      class: d(k)(
        "h-5 w-5 border-2 p-[3px] bg-clip-content rounded-full appearance-none cursor-pointer focus-visible:outline focus-visible:outline-offset disabled:border-disabled-500 disabled:cursor-not-allowed disabled:checked:bg-disabled-500 disabled:checked:border-disabled-500",
        e.invalid && !e.disabled ? "border-negative-700 checked:bg-negative-700 hover:border-negative-800 hover:checked:bg-negative-800 active:border-negative-900 active:checked:bg-negative-900" : "border-neutral-500 active:border-primary-900 hover:border-primary-700 checked:bg-primary-700 checked:border-primary-700 hover:checked:bg-primary-800 hover:checked:border-primary-800 active:checked:bg-primary-900 active:checked:border-primary-900",
        a.$attrs.class
      ),
      disabled: e.disabled
    }, d(n)), null, 16, w)), [
      [h, r.value]
    ]);
  }
});
export {
  R as default
};
