// @storefront-ui/vue/dist/components/SfTooltip/SfTooltip.vue.mjs
import { defineComponent as S, toRefs as k, watch as i, openBlock as a, createElementBlock as l, mergeProps as r, unref as o, renderSlot as V, createTextVNode as v, toDisplayString as B, createCommentVNode as s } from "vue";
import { useTooltip as P } from "../../composables/useTooltip/useTooltip.mjs";
const A = ["id"], q = /* @__PURE__ */ S({
  __name: "SfTooltip",
  props: {
    placement: {
      type: String,
      default: void 0
    },
    middleware: {
      type: Array,
      default: void 0
    },
    strategy: {
      type: String,
      default: void 0
    },
    showArrow: {
      type: Boolean,
      default: !1
    },
    label: {
      type: String,
      required: !0
    },
    id: {
      type: String,
      default: void 0
    },
    modelValue: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: n }) {
    const p = e, m = n, { placement: u, middleware: c, strategy: f, modelValue: y } = k(p), { isOpen: d, close: g, open: w, triggerProps: x, tooltipProps: b, arrowProps: h } = P({ placement: u, middleware: c, strategy: f });
    return i(
      y,
      (t) => {
        t ? w() : g();
      },
      { immediate: !0 }
    ), i(d, (t) => {
      m("update:modelValue", t);
    }), (t, T) => (a(), l("span", r({ "data-testid": "tooltip" }, o(x)), [
      V(t.$slots, "default"),
      e.label && o(d) ? (a(), l("div", r({
        key: 0,
        id: e.id,
        role: "tooltip",
        class: "bg-black px-2 py-1.5 rounded-md text-white text-xs w-max max-w-[360px] drop-shadow-sm"
      }, o(b)), [
        v(B(e.label) + " ", 1),
        e.showArrow ? (a(), l("span", r({ key: 0 }, o(h), { class: "rotate-45 bg-black" }), null, 16)) : s("", !0)
      ], 16, A)) : s("", !0)
    ], 16));
  }
});
export {
  q as default
};
