// @storefront-ui/vue/dist/components/SfTextarea/SfTextarea.vue.mjs
import { defineComponent as f, toRefs as d, computed as g, withDirectives as x, openBlock as v, createElementBlock as y, mergeProps as h, unref as a, vModelText as b } from "vue";
import { useTwMergeRoot as S, twMerge as V } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfTextareaSize as t } from "@storefront-ui/shared";
import { useFocusVisible as T } from "../../composables/useFocusVisible/useFocusVisible.mjs";
const w = {
  [t.sm]: " h-[56px] py-[6px] pl-4 pr-3",
  [t.base]: "h-[64px] py-2 pl-4 pr-3",
  [t.lg]: "h-[72px], p-3 pl-4"
}, k = /* @__PURE__ */ f({
  inheritAttrs: !1,
  __name: "SfTextarea",
  props: {
    modelValue: {
      type: [String, Number],
      default: ""
    },
    size: {
      type: String,
      default: t.base
    },
    invalid: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: o }) {
    const l = e, u = o, { attrsWithoutClass: p } = S(), { modelValue: r } = d(l), { isFocusVisible: m } = T({ isTextInput: !0 }), s = g({
      get: () => r == null ? void 0 : r.value,
      set: (i) => u("update:modelValue", i)
    });
    return (i, n) => x((v(), y("textarea", h({
      "onUpdate:modelValue": n[0] || (n[0] = (c) => s.value = c),
      class: a(V)(
        "px-4 bg-white rounded-xl text-neutral-900 ring-inset hover:ring-primary-800 focus:caret-primary-700 active:caret-primary-700 active:ring-primary-700 active:ring-2 focus:ring-primary-700 focus:ring-2 outline-hidden",
        {
          "ring-2 ring-negative-700": e.invalid,
          "ring-1 ring-neutral-300": !e.invalid,
          "focus:outline focus:outline-offset": a(m)
        },
        w[e.size],
        i.$attrs.class
      ),
      "data-testid": "textarea"
    }, a(p)), null, 16)), [
      [b, s.value]
    ]);
  }
});
export {
  k as default
};
