// @storefront-ui/vue/dist/components/SfSwitch/SfSwitch.vue.mjs
import { defineComponent as u, toRefs as h, ref as p, computed as m, withDirectives as g, openBlock as v, createElementBlock as k, mergeProps as y, unref as c, vModelCheckbox as w } from "vue";
import { useTwMergeRoot as x, twMerge as B } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
const C = ["value", "aria-checked"], M = /* @__PURE__ */ u({
  inheritAttrs: !1,
  __name: "SfSwitch",
  props: {
    modelValue: {
      type: [String, Array, Boolean],
      default: !1
    },
    value: {
      type: String,
      default: ""
    },
    invalid: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: b }) {
    const l = e, { attrsWithoutClass: s } = x(), { modelValue: r } = h(l), n = b, t = p(), a = m({
      get: () => r == null ? void 0 : r.value,
      set: (o) => n("update:modelValue", o)
    });
    return (o, i) => {
      var d;
      return g((v(), k("input", y({
        ref_key: "switchRef",
        ref: t,
        "onUpdate:modelValue": i[0] || (i[0] = (f) => a.value = f),
        class: c(B)(
          "appearance-none h-5 min-w-[36px] bg-transparent border-2 border-gray-500 rounded-full relative ease-in-out duration-300 hover:border-primary-800 hover:checked:before:bg-white checked:before:left-1/2 checked:before:ml-0 checked:before:mr-0.5 disabled:before:bg-gray-500/50 hover:before:bg-primary-800 active:border-primary-800 active:before:bg-primary-800 checked:bg-none checked:bg-primary-700 checked:border-primary-700 checked:before:bg-white hover:checked:bg-primary-800 hover:checked:border-primary-800 disabled:border-gray-500/50 checked:disabled:before:bg-white checked:disabled:bg-gray-500/50 checked:disabled:border-0 before:transition-all  before:w-3.5 before:h-3.5 before:bg-gray-500 before:absolute before:top-0 before:bottom-0 before:my-auto before:rounded-full before:left-0 before:ml-0.5 before:ease-in-out before:duration-300 cursor-pointer disabled:cursor-not-allowed focus-visible:outline focus-visible:outline-offset",
          {
            "border-negative-700 hover:border-negative-800 active:border-negative-900 before:bg-negative-900": e.invalid
          },
          o.$attrs.class
        ),
        value: e.value,
        type: "checkbox",
        role: "switch",
        "data-testid": "switch",
        "aria-checked": (d = t.value) == null ? void 0 : d.checked
      }, c(s)), null, 16, C)), [
        [w, a.value]
      ]);
    };
  }
});
export {
  M as default
};
