// @storefront-ui/vue/dist/components/SfChip/SfChip.vue.mjs
import { defineComponent as w, toRefs as $, useId as z, computed as f, openBlock as V, createElementBlock as B, Fragment as P, withDirectives as M, createElementVNode as m, mergeProps as g, unref as t, vModelCheckbox as R, renderSlot as a, createCommentVNode as x } from "vue";
import { useTwMergeRoot as j, twMerge as b } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { useSlotsRef as q } from "../../shared/reactiveContext.mjs";
import { SfChipSize as s } from "@storefront-ui/shared";
const A = ["id"], E = ["for"], F = {
  [s.sm]: "text-sm py-1.5 gap-1.5",
  [s.base]: "text-base py-2 gap-2"
}, T = /* @__PURE__ */ w({
  inheritAttrs: !1,
  __name: "SfChip",
  props: {
    size: {
      type: String,
      default: s.base
    },
    modelValue: {
      type: [String, Array, Boolean],
      default: !1
    },
    inputProps: {
      type: Object,
      default: null
    },
    square: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(i, { emit: v }) {
    const y = i, { attrsWithoutClass: h } = j(), { size: o, square: l, modelValue: n } = $(y), k = v, r = q(), p = z(), u = f({
      get: () => n == null ? void 0 : n.value,
      set: (e) => k("update:modelValue", e)
    }), S = f(() => {
      switch (o.value) {
        case s.sm:
          return l.value ? ["px-1.5"] : [r.value.prefix ? "pl-1.5" : "pl-3", r.value.suffix ? "pr-1.5" : "pr-3"];
        default:
          return l.value ? ["px-2"] : [r.value.prefix ? "pl-2" : "pl-4", r.value.suffix ? "pr-2" : "pr-4"];
      }
    });
    return (e, d) => {
      var c;
      return V(), B(P, null, [
        M(m("input", g({
          id: t(p),
          "onUpdate:modelValue": d[0] || (d[0] = (C) => u.value = C),
          class: t(b)("absolute w-0 outline-hidden appearance-none peer", (c = i.inputProps) == null ? void 0 : c.class),
          type: "checkbox"
        }, i.inputProps), null, 16, A), [
          [R, u.value]
        ]),
        m("label", g({
          for: t(p),
          class: t(b)(
            "cursor-pointer ring-1 ring-neutral-200 ring-inset rounded-full inline-flex items-center transition duration-300 justify-center outline-offset-2 outline-secondary-600 peer-next-checked:ring-2 peer-next-checked:ring-primary-700 hover:bg-primary-100 peer-next-hover:ring-primary-200 active:bg-primary-200 peer-next-active:ring-primary-300 peer-next-disabled:cursor-not-allowed peer-next-disabled:bg-disabled-100 peer-next-disabled:opacity-50 peer-next-disabled:ring-1 peer-next-disabled:ring-disabled-200 peer-next-disabled:hover:ring-disabled-200 peer-next-checked:hover:ring-primary-700 peer-next-checked:active:ring-primary-700 peer-next-focus-visible:outline",
            F[t(o)],
            ...S.value,
            e.$attrs.class
          ),
          "data-testid": "chip"
        }, t(h)), [
          e.$slots.prefix ? a(e.$slots, "prefix", { key: 0 }) : x("", !0),
          a(e.$slots, "default"),
          e.$slots.suffix ? a(e.$slots, "suffix", { key: 1 }) : x("", !0)
        ], 16, E)
      ], 64);
    };
  }
});
export {
  T as default
};
