// @grapesjs/studio-sdk-plugins/dist/animationComponent/index.umd.js
(function(h,b){typeof exports=="object"&&typeof module<"u"?module.exports=b():typeof define=="function"&&define.amd?define(b):(h=typeof globalThis<"u"?globalThis:h||self,h.StudioSdkPlugins_animationComponent=b())})(this,function(){"use strict";const h="app.grapesjs.com",b="app-stage.grapesjs.com",A=[h,"app2.grapesjs.com",b,"app-stage2.grapesjs.com","localhost","127.0.0.1",".local-credentialless.webcontainer.io",".local.webcontainer.io","-sandpack.codesandbox.io"],O="license:check:start",G="license:check:end",V=()=>typeof window<"u",B=({isDev:t,isStage:n})=>`${t?"":`https://${n?b:h}`}/api`,M=()=>{const t=V()&&window.location.hostname;return!!t&&(A.includes(t)||A.some(n=>t.endsWith(n)))};async function F({path:t,baseApiUrl:n,method:e="GET",headers:r={},params:i,body:a}){const c=`${n||B({isDev:!1,isStage:!1})}${t}`,l={method:e,headers:{"Content-Type":"application/json",...r}};a&&(l.body=JSON.stringify(a));const f=i?new URLSearchParams(i).toString():"",s=f?`?${f}`:"",o=await fetch(`${c}${s}`,l);if(!o.ok)throw new Error(`HTTP error! status: ${o.status}`);return o.json()}var I=(t=>(t.free="free",t.startup="startup",t.business="business",t.enterprise="enterprise",t))(I||{});const U={[I.free]:0,[I.startup]:10,[I.business]:20,[I.enterprise]:30};function X(t){const n=t;return n.init=e=>r=>t(r,e),n}const _=t=>X(t);async function j({editor:t,plan:n,pluginName:e,licenseKey:r,cleanup:i}){let a="",m=!1;const c=M(),l=s=>{console.warn("Cleanup plugin:",e,"Reason:",s),i()},f=(s={})=>{var u;const{error:o,sdkLicense:p}=s,d=(u=s.plan)==null?void 0:u.category;if(!(p||s.license)||o)l(o||"Invalid license");else if(d){const Z=U[n],N=U[d];Z>N&&l({pluginRequiredPlan:n,licensePlan:d})}};t.on(O,s=>{a=s==null?void 0:s.baseApiUrl,m=!0}),t.on(G,s=>{f(s)}),setTimeout(async()=>{if(!m){if(c)return;if(r){const s=await W({licenseKey:r,pluginName:e,baseApiUrl:a});s&&f(s)}else l("The `licenseKey` option not provided")}},2e3)}async function W(t){const{licenseKey:n,pluginName:e,baseApiUrl:r}=t;try{return(await F({baseApiUrl:r,path:`/sdk/${n||"na"}`,method:"POST",params:{d:window.location.hostname,pn:e}})).result||{}}catch(i){return console.error("Error during SDK license check:",i),!1}}const Y=(...t)=>n=>t.some(e=>n.is(e));var y=(t=>(t.name="animation-name",t.duration="animation-duration",t.delay="animation-delay",t.iterationCount="animation-iteration-count",t.timingFunction="animation-timing-function",t.fillMode="animation-fill-mode",t))(y||{}),v=(t=>(t.onScroll="--animation-on-scroll",t.threshold="--animation-threshold",t.repeat="--animation-repeat",t))(v||{});const g="animation",S="animation-group",L="animation:update",E="animation-group:update",D={category:{id:"animations",label:"Animations"},select:!0},x=t=>(t==null?void 0:t.getType())===g,C=t=>(t==null?void 0:t.getType())===S,R=t=>x(t)||C(t),w={isVisible:({component:t})=>x(t)},$={isVisible:({component:t})=>C(t)},H={isVisible:({component:t})=>x(t)&&C(t==null?void 0:t.parent())?!1:R(t)},T={isVisible:({component:t,sector:n})=>{var e;return x(t)&&C(t==null?void 0:t.parent())?!1:R(t)&&((e=n.getProperty(v.onScroll))==null?void 0:e.getValue())==="true"}},P={animation:'<svg viewBox="0 0 24 24"><path fill="currentColor" d="M16.5 14a6.5 6.5 0 1 0 0-13 6.5 6.5 0 0 0 0 13"/><path fill="currentColor" d="M17.14 15.98a8.5 8.5 0 0 1-9.12-9.12 6.5 6.5 0 1 0 9.12 9.12"/><path fill="currentColor" d="M12.64 20.48a8.5 8.5 0 0 1-9.12-9.12 6.5 6.5 0 1 0 9.12 9.12"/></svg>',animationGroup:'<svg viewBox="0 0 24 24"><path d="M4 2a2 2 0 0 0-2 2v10h2V4h10V2H4m4 4a2 2 0 0 0-2 2v10h2V8h10V6H8m4 4a2 2 0 0 0-2 2v8c0 1.1.9 2 2 2h8a2 2 0 0 0 2-2v-8a2 2 0 0 0-2-2h-8Z"/></svg>'},K=function(t){const n=this;let e=null;const r=()=>{e==null||e.disconnect(),e=null},i=()=>{r();const a=n.style;a.animationName="",requestAnimationFrame(()=>{const m=getComputedStyle(n),c=m.getPropertyValue("animation-name").trim(),l=m.getPropertyValue("--animation-on-scroll").trim()!=="false",f=parseFloat(m.getPropertyValue("--animation-threshold").trim())||.2,s=m.getPropertyValue("--animation-repeat").trim()==="true",o=()=>{a.animationName="none",n.offsetWidth,a.animationName=c,a.animationPlayState="running"};if(!c||c==="none"){a.animationName="none",a.animationPlayState="paused";return}if(!l){o();return}a.animationPlayState="paused",e=new IntersectionObserver(p=>{p.forEach(d=>{d.isIntersecting&&(o(),!s&&r())})},{threshold:f}),e.observe(n)})};n.addEventListener(t.animationUpdateEvent,i),i()},q=(t,n)=>{const{Components:e,Blocks:r,Canvas:i}=t,{block:a={},animationList:m=[]}=n,c=t.Components.events,l="Animation",f=Object.values(v),s=Object.values(y);return e.addType(g,{block:a?{label:l,media:P.animation,content:{type:g},...D,...a}:void 0,model:{defaults:{emptyState:!0,name:l,traits:[],script:K,animationUpdateEvent:L,"script-props":["animationUpdateEvent"],styles:m.map(o=>o.css).join(`
`)},init(){const{em:o}=this;this.addStyle({[y.name]:"fadeIn",[y.duration]:"1s",[y.fillMode]:"both",...n.animationStyle,...this.getStyle()}),this.listenTo(o,c.mount,p=>{var k;if(p!==this)return;((k=this.parent())==null?void 0:k.get("type"))!==S||f.forEach(u=>this.removeStyle(u))})}},view:{events:()=>({animationstart:"onAnimationStart",animationend:"onAnimationEnd"}),init(){const{model:o}=this;this.listenTo(o,c.styleUpdate,this.handleStyleChange)},onAnimationStart(){var p;if((p=this.cmpSpots)!=null&&p.length)return;const o=i.getSpots({component:this.model});this.cmpSpots=o,i.removeSpots(o),this.el.ownerDocument.body.style.overflowX="hidden"},onAnimationEnd(){const o=this.cmpSpots||[];i.spots.add(o),this.cmpSpots=[],this.el.ownerDocument.body.style.overflowX=""},triggerAnimation(){this.el.dispatchEvent(new CustomEvent(L))},handleStyleChange(o,p){Object.keys((p==null?void 0:p.style)||{}).some(u=>s.includes(u)||f.includes(u))&&this.triggerAnimation()},onRender(){this.triggerAnimation()}}}),()=>{r.remove(g),e.removeType(g)}},J=function(t){const n=this,e="--animation-on-scroll",r="--animation-threshold",i="--animation-repeat";let a=null;const m=()=>{a==null||a.disconnect(),a=null},c=()=>{m();const l=getComputedStyle(n),f=l.getPropertyValue(e).trim()!=="false",s=parseFloat(l.getPropertyValue(r).trim())||.2,o=l.getPropertyValue(i).trim()==="true",p=()=>{Array.from(n.children).forEach(d=>{d instanceof HTMLElement&&typeof d.dispatchEvent=="function"&&d.dispatchEvent(new CustomEvent(E,{bubbles:!1}))})};if(!f){p();return}a=new IntersectionObserver(d=>{d.forEach(k=>{k.isIntersecting?(p(),!o&&m()):o&&Array.from(n.children).forEach(u=>{u instanceof HTMLElement&&getComputedStyle(u).getPropertyValue(e).trim()!=="false"&&(u.style.opacity="0",u.style.animationPlayState="paused")})})},{threshold:s}),a.observe(n)};n.addEventListener(t.animationGroupUpdateEvent,c),c()},Q=(t,n)=>{const{Components:e,Blocks:r}=t,{blockGroup:i={}}=n,a="Animation Group";return e.addType(S,{block:i?{label:a,media:P.animationGroup,content:{type:S,components:[{type:g},{type:g},{type:g}]},...D,...i}:void 0,model:{defaults:{name:a,emptyState:!0,traits:[],script:J,animationGroupUpdateEvent:E,"script-props":["animationGroupUpdateEvent"],droppable:Y(g)},init(){this.addStyle({...n.animationGroupStyle,...this.getStyle()});const m=this.components();this.listenTo(m,"add remove",this.updateChildrenStaggerStyles),this.updateChildrenStaggerStyles()},updateChildrenStaggerStyles(){this.components().forEach((m,c)=>{m.is(g)&&m.addStyle({"--stagger-index":c.toString(),[y.delay]:"calc(var(--stagger-index, 0) * var(--stagger-delay, 0.3s))"},{partial:!0})})}},view:{init(){const{model:m}=this;this.listenTo(m,"component:styleUpdate",this.handleStyleChange)},triggerAnimation(){this.el.dispatchEvent(new CustomEvent(E))},handleStyleChange(){this.triggerAnimation(),this.model.updateChildrenStaggerStyles()},onRender(){this.triggerAnimation()}}}),()=>{r.remove(S),e.removeType(S)}},tt="animationComponent",nt=I.startup,z=[{id:"flash",name:"Flash",css:`@keyframes flash {
      0%, 50%, to {
          opacity:1
      }

      25%, 75% {
          opacity:0
      }
  }`},{id:"pulse",name:"Pulse",css:`@keyframes pulse {
      0% {
          transform:scaleX(1)
      }

      50% {
          transform:scale3d(1.05, 1.05, 1.05)
      }

      to {
          transform:scaleX(1)
      }
  }`},{id:"shake",name:"Shake",css:`@keyframes shake {
      0%, to {
          transform:translateZ(0)
      }

      10%, 30%, 50%, 70%, 90% {
          transform:translate3d(-10px, 0, 0)
      }

      20%, 40%, 60%, 80% {
          transform:translate3d(10px, 0, 0)
      }
  }`},{id:"tada",name:"Tada",css:`@keyframes tada {
      0% {
          transform:scaleX(1)
      }

      10%, 20% {
          transform:scale3d(.9, .9, .9) rotate(-3deg)
      }

      30%, 50%, 70%, 90% {
          transform:scale3d(1.1, 1.1, 1.1) rotate(3deg)
      }

      40%, 60%, 80% {
          transform:scale3d(1.1, 1.1, 1.1) rotate(-3deg)
      }

      to {
          transform:scaleX(1)
      }
  }`},{id:"heartBeat",name:"Heart Beat",css:`@keyframes heartBeat {
      0% {
          transform:scale(1)
      }

      14% {
          transform:scale(1.3)
      }

      28% {
          transform:scale(1)
      }

      42% {
          transform:scale(1.3)
      }

      70% {
          transform:scale(1)
      }
  }`},{id:"bounce",name:"Bounce",css:`@keyframes bounce {
      0%, 20%, 53%, 80%, to {
          animation-timing-function: cubic-bezier(.215, .61, .355, 1);
          transform:translateZ(0)
      }

      40%, 43% {
          animation-timing-function: cubic-bezier(.755, .05, .855, .06);
          transform:translate3d(0, -30px, 0)
      }

      70% {
          animation-timing-function: cubic-bezier(.755, .05, .855, .06);
          transform:translate3d(0, -15px, 0)
      }

      90% {
          transform:translate3d(0, -4px, 0)
      }
  }`},{id:"bounceIn",name:"Bounce In",css:`@keyframes bounceIn {
      0%, 20%, 40%, 60%, 80%, to {
          animation-timing-function:cubic-bezier(.215, .61, .355, 1)
      }

      0% {
          opacity: 0;
          transform:scale3d(.3, .3, .3)
      }

      20% {
          transform:scale3d(1.1, 1.1, 1.1)
      }

      40% {
          transform:scale3d(.9, .9, .9)
      }

      60% {
          opacity: 1;
          transform:scale3d(1.03, 1.03, 1.03)
      }

      80% {
          transform:scale3d(.97, .97, .97)
      }

      to {
          opacity: 1;
          transform:scaleX(1)
      }
  }`},{id:"bounceInDown",name:"Bounce In Down",css:`@keyframes bounceInDown {
      0%, 60%, 75%, 90%, to {
          animation-timing-function:cubic-bezier(.215, .61, .355, 1)
      }

      0% {
          opacity: 0;
          transform:translate3d(0, -100%, 0)
      }

      60% {
          opacity: 1;
          transform:translate3d(0, 25px, 0)
      }

      75% {
          transform:translate3d(0, -10px, 0)
      }

      90% {
          transform:translate3d(0, 5px, 0)
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"bounceInLeft",name:"Bounce In Left",css:`@keyframes bounceInLeft {
      0%, 1%, 60%, 75%, 90%, to {
          animation-timing-function:cubic-bezier(.215, .61, .355, .1)
      }

      0% {
          opacity: 0;
      }

      1% {
          transform:translate3d(-100%, 0, 0)
      }

      60% {
          opacity: 1;
          transform:translate3d(25px, 0, 0)
      }

      75% {
          transform:translate3d(-10px, 0, 0)
      }

      90% {
          transform:translate3d(5px, 0, 0)
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"bounceInRight",name:"Bounce In Right",css:`@keyframes bounceInRight {
      0%, 1%, 60%, 75%, 90%, to {
          animation-timing-function:cubic-bezier(.215, .61, .355, .1)
      }

      0% {
          opacity: 0;
      }

      1% {
          transform:translate3d(100%, 0, 0)
      }

      60% {
          opacity: 1;
          transform:translate3d(-25px, 0, 0)
      }

      75% {
          transform:translate3d(10px, 0, 0)
      }

      90% {
          transform:translate3d(-5px, 0, 0)
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"bounceInUp",name:"Bounce In Up",css:`@keyframes bounceInUp {
      0%, 1%, 60%, 75%, 90%, to {
          animation-timing-function:cubic-bezier(.215, .61, .355, .1)
      }

      0% {
          opacity: 0;
      }

      1% {
          transform:translate3d(0, 100%, 0)
      }

      60% {
          opacity: 1;
          transform:translate3d(0, -20px, 0)
      }

      75% {
          transform:translate3d(0, 10px, 0)
      }

      90% {
          transform:translate3d(0, -5px, 0)
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"fadeIn",name:"Fade In",css:`@keyframes fadeIn {
      0% {
          opacity:0
      }

      to {
          opacity:1
      }
  }`},{id:"fadeInDown",name:"Fade In Down",css:`@keyframes fadeInDown {
      0% {
          opacity: 0;
      }

      1% {
          transform:translate3d(0, -100%, 0)
      }

      to {
          opacity: 1;
          transform:translateZ(0)
      }
  }`},{id:"fadeInLeft",name:"Fade In Left",css:`@keyframes fadeInLeft {
      0% {
          opacity: 0;
      }

      1% {
          transform:translate3d(-100%, 0, 0)
      }

      to {
          opacity: 1;
          transform:translateZ(0)
      }
  }`},{id:"fadeInRight",name:"Fade In Right",css:`@keyframes fadeInRight {
      0% {
          opacity: 0;
      }

      1% {
          transform: translate3d(90%, 0, 0)
      }

      to {
          opacity: 1;
          transform:translateZ(0)
      }
  }`},{id:"fadeInUp",name:"Fade In Up",css:`@keyframes fadeInUp {
      0% {
          opacity: 0;
      }

      1% {
          transform:translate3d(0, 100%, 0)
      }

      to {
          opacity: 1;
          transform:translateZ(0)
      }
  }`},{id:"flipInX",name:"Flip In X",css:`@keyframes flipInX {
      0% {
          transform: perspective(400px) rotateX(90deg);
          animation-timing-function: ease-in;
          opacity:0
      }

      40% {
          transform: perspective(400px) rotateX(-20deg);
          animation-timing-function:ease-in
      }

      60% {
          transform: perspective(400px) rotateX(10deg);
          opacity:1
      }

      80% {
          transform:perspective(400px) rotateX(-5deg)
      }

      to {
          transform:perspective(400px)
      }
  }`},{id:"flipInY",name:"Flip In Y",css:`@keyframes flipInY {
      0% {
          transform: perspective(400px) rotateY(90deg);
          animation-timing-function: ease-in;
          opacity:0
      }

      40% {
          transform: perspective(400px) rotateY(-20deg);
          animation-timing-function:ease-in
      }

      60% {
          transform: perspective(400px) rotateY(10deg);
          opacity:1
      }

      80% {
          transform:perspective(400px) rotateY(-5deg)
      }

      to {
          transform:perspective(400px)
      }
  }`},{id:"rotateIn",name:"Rotate In",css:`@keyframes rotateIn {
      0% {
          transform-origin: center;
          transform: rotate(-200deg);
          opacity:0
      }

      to {
          transform-origin: center;
          transform: translateZ(0);
          opacity:1
      }
  }`},{id:"rotateInDownLeft",name:"Rotate In Down Left",css:`@keyframes rotateInDownLeft {
      0% {
          transform-origin: left bottom;
          transform: rotate(-45deg);
          opacity:0
      }

      to {
          transform-origin: left bottom;
          transform: translateZ(0);
          opacity:1
      }
  }`},{id:"rotateInDownRight",name:"Rotate In Down Right",css:`@keyframes rotateInDownRight {
      0% {
          transform-origin: right bottom;
          transform: rotate(45deg);
          opacity:0
      }

      to {
          transform-origin: right bottom;
          transform: translateZ(0);
          opacity:1
      }
  }`},{id:"rotateInUpLeft",name:"Rotate In Up Left",css:`@keyframes rotateInUpLeft {
      0% {
          opacity: 0
      }

      1% {
          transform-origin: left bottom;
          transform: rotate(45deg);
      }

      to {
          transform-origin: left bottom;
          transform: translateZ(0);
          opacity:1
      }
  }`},{id:"rotateInUpRight",name:"Rotate In Up Right",css:`@keyframes rotateInUpRight {
      0% {
          opacity: 0
      }

      1% {
          transform-origin: right bottom;
          transform: rotate(-90deg);
      }

      to {
          transform-origin: right bottom;
          transform: translateZ(0);
          opacity:1
      }
  }`},{id:"slideInDown",name:"Slide In Down",css:`@keyframes slideInDown {
      0% {
          visibility:visible
      }

      1% {
          transform: translate3d(0, -100%, 0);
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"slideInLeft",name:"Slide In Left",css:`@keyframes slideInLeft {
      0% {
          visibility:visible
      }

      1% {
          transform: translate3d(-100%, 0, 0);
      }

      to {
          transform: translateZ(0)
      }
  }`},{id:"slideInRight",name:"Slide In Right",css:`@keyframes slideInRight {
      0% {
          visibility:visible
      }

      1% {
          transform: translate3d(100%, 0, 0);
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"slideInUp",name:"Slide In Up",css:`@keyframes slideInUp {
      0% {
          visibility:visible
      }

      1% {
          transform: translate3d(0, 100%, 0);
      }

      to {
          transform:translateZ(0)
      }
  }`},{id:"zoomIn",name:"Zoom In",css:`@keyframes zoomIn {
      0% {
          opacity: 0;
          transform:scale3d(.3, .3, .3)
      }

      50% {
          opacity:1
      }
  }`},{id:"zoomInDown",name:"Zoom In Down",css:`@keyframes zoomInDown {
      0% {
          opacity: 0;
          transform: scale3d(.1, .1, .1) translate3d(0, -1000px, 0);
          animation-timing-function:cubic-bezier(.55, .055, .675, .19)
      }

      60% {
          opacity: 1;
          transform: scale3d(.475, .475, .475) translate3d(0, 60px, 0);
          animation-timing-function:cubic-bezier(.175, .885, .32, 1)
      }
  }`},{id:"zoomInLeft",name:"Zoom In Left",css:`@keyframes zoomInLeft {
      0% {
          opacity: 0;
          transform: scale3d(.1, .1, .1) translate3d(-1000px, 0, 0);
          animation-timing-function:cubic-bezier(.55, .055, .675, .19)
      }

      60% {
          opacity: 1;
          transform: scale3d(.475, .475, .475) translate3d(10px, 0, 0);
          animation-timing-function:cubic-bezier(.175, .885, .32, 1)
      }
  }`},{id:"zoomInRight",name:"Zoom In Right",css:`@keyframes zoomInRight {
      0% {
          opacity: 0;
          transform: scale3d(.1, .1, .1) translate3d(1000px, 0, 0);
          animation-timing-function:cubic-bezier(.55, .055, .675, .19)
      }

      60% {
          opacity: 1;
          transform: scale3d(.475, .475, .475) translate3d(-10px, 0, 0);
          animation-timing-function:cubic-bezier(.175, .885, .32, 1)
      }
  }`},{id:"zoomInUp",name:"Zoom In Up",css:`@keyframes zoomInUp {
      0% {
          opacity: 0;
          animation-timing-function: cubic-bezier(.55, .055, .675, .19);
      }

      1% {
          transform: scale3d(.1, .1, .1) translate3d(0, 100%, 0);
      }

      60% {
          opacity: 1;
          transform: scale3d(.475, .475, .475) translate3d(0, -60px, 0);
          animation-timing-function:cubic-bezier(.175, .885, .32, 1)
      }
  }`}];return _(function(t,n={}){var i;const e=((i=n.animations)==null?void 0:i.call(n,{items:z}))||z,r=[q(t,{...n,animationList:e}),Q(t,n)];t.Styles.addSector("animation",{name:"Animation",properties:[{...w,name:"Type",property:y.name,type:"select",full:!0,default:"fadeIn",options:e},{...w,name:"Duration",property:y.duration,type:"number",units:["s"],min:.1,max:10,step:.1,default:"1s"},{...w,name:"Delay",property:y.delay,type:"number",units:["s"],min:0,max:10,step:.1,default:"1s"},{...w,name:"Iteration Count",property:y.iterationCount,type:"select",default:"1",options:[{id:"1",name:"One time"},{id:"infinite",name:"Infinite"}]},{...w,name:"Timing",property:y.timingFunction,type:"select",default:"ease",options:[{id:"ease",name:"Ease"},{id:"ease-in",name:"Ease in"},{id:"ease-out",name:"Ease out"},{id:"ease-in-out",name:"Ease in out"},{id:"linear",name:"Linear"}]},{...$,name:"Stagger Delay",property:"--stagger-delay",type:"slider",units:["s"],min:.1,max:5,step:.1,default:"0.3s"},{...H,name:"Animate on Scroll",property:v.onScroll,type:"radio",default:"true",options:[{name:"Yes",id:"true"},{name:"No",id:"false"}]},{...T,name:"Scroll Threshold",property:v.threshold,type:"number",full:!0,min:.1,max:1,step:.01,default:"0.2"},{...T,name:"Repeat on Scroll",property:v.repeat,type:"radio",default:"false",options:[{name:"Yes",id:"true"},{name:"No",id:"false"}]}]}),j({editor:t,licenseKey:n.licenseKey,plan:nt,pluginName:tt,cleanup:()=>{r.forEach(a=>a())}})})});
