// @polar-sh/sdk/dist/commonjs/models/operations/walletslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletsListResponse$inboundSchema = exports.WalletsListRequest$outboundSchema = exports.WalletsListQueryParamCustomerIDFilter$outboundSchema = exports.WalletsListQueryParamOrganizationIDFilter$outboundSchema = void 0;
exports.walletsListQueryParamOrganizationIDFilterToJSON = walletsListQueryParamOrganizationIDFilterToJSON;
exports.walletsListQueryParamCustomerIDFilterToJSON = walletsListQueryParamCustomerIDFilterToJSON;
exports.walletsListRequestToJSON = walletsListRequestToJSON;
exports.walletsListResponseFromJSON = walletsListResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourcewallet_js_1 = require("../components/listresourcewallet.js");
const walletsortproperty_js_1 = require("../components/walletsortproperty.js");
/** @internal */
exports.WalletsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
function walletsListQueryParamOrganizationIDFilterToJSON(walletsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(exports.WalletsListQueryParamOrganizationIDFilter$outboundSchema.parse(walletsListQueryParamOrganizationIDFilter));
}
/** @internal */
exports.WalletsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
function walletsListQueryParamCustomerIDFilterToJSON(walletsListQueryParamCustomerIDFilter) {
    return JSON.stringify(exports.WalletsListQueryParamCustomerIDFilter$outboundSchema.parse(walletsListQueryParamCustomerIDFilter));
}
/** @internal */
exports.WalletsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(walletsortproperty_js_1.WalletSortProperty$outboundSchema))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
    });
}));
function walletsListRequestToJSON(walletsListRequest) {
    return JSON.stringify(exports.WalletsListRequest$outboundSchema.parse(walletsListRequest));
}
/** @internal */
exports.WalletsListResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcewallet_js_1.ListResourceWallet$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function walletsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WalletsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WalletsListResponse' from JSON`);
}
