// @polar-sh/sdk/dist/commonjs/models/operations/webhookslistwebhookdeliveries.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksListWebhookDeliveriesResponse$inboundSchema = exports.WebhooksListWebhookDeliveriesRequest$outboundSchema = exports.EndpointId$outboundSchema = void 0;
exports.endpointIdToJSON = endpointIdToJSON;
exports.webhooksListWebhookDeliveriesRequestToJSON = webhooksListWebhookDeliveriesRequestToJSON;
exports.webhooksListWebhookDeliveriesResponseFromJSON = webhooksListWebhookDeliveriesResponseFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const listresourcewebhookdelivery_js_1 = require("../components/listresourcewebhookdelivery.js");
/** @internal */
exports.EndpointId$outboundSchema = z.union([z.string(), z.array(z.string())]);
function endpointIdToJSON(endpointId) {
    return JSON.stringify(exports.EndpointId$outboundSchema.parse(endpointId));
}
/** @internal */
exports.WebhooksListWebhookDeliveriesRequest$outboundSchema = z.pipe(z.object({
    endpointId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    startTimestamp: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    endTimestamp: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        endpointId: "endpoint_id",
        startTimestamp: "start_timestamp",
        endTimestamp: "end_timestamp",
    });
}));
function webhooksListWebhookDeliveriesRequestToJSON(webhooksListWebhookDeliveriesRequest) {
    return JSON.stringify(exports.WebhooksListWebhookDeliveriesRequest$outboundSchema.parse(webhooksListWebhookDeliveriesRequest));
}
/** @internal */
exports.WebhooksListWebhookDeliveriesResponse$inboundSchema = z.pipe(z.object({
    Result: listresourcewebhookdelivery_js_1.ListResourceWebhookDelivery$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
}));
function webhooksListWebhookDeliveriesResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhooksListWebhookDeliveriesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookDeliveriesResponse' from JSON`);
}
