// @storefront-ui/vue/dist/components/SfSelect/SfSelect.vue.mjs
import { defineComponent as V, toRefs as h, ref as C, computed as z, openBlock as c, createElementBlock as f, normalizeClass as r, unref as e, withDirectives as B, createElementVNode as N, mergeProps as k, withKeys as q, toDisplayString as M, createCommentVNode as $, renderSlot as p, vModelSelect as D, createVNode as E } from "vue";
import { useTwMergeRoot as F, twMerge as g } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import K from "../SfIcons/SfIconExpandMore.vue.mjs";
import { SfSelectSize as i } from "@storefront-ui/shared";
import { useDisclosure as R } from "../../composables/useDisclosure/useDisclosure.mjs";
import { useFocusVisible as A } from "../../composables/useFocusVisible/useFocusVisible.mjs";
const O = ["required", "disabled"], H = /* @__PURE__ */ V({
  inheritAttrs: !1,
  __name: "SfSelect",
  props: {
    size: {
      type: String,
      default: i.base
    },
    placeholder: {
      type: String,
      default: ""
    },
    required: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    modelValue: {
      type: String,
      default: ""
    },
    wrapperClassName: {
      type: String,
      default: ""
    }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: b }) {
    const y = t, v = b, { attrsWithoutClass: S } = F(), { modelValue: d } = h(y), { isOpen: x, close: n, open: o } = R(), { isFocusVisible: w } = A(), u = C(d.value), m = z({
      get: () => d.value || u.value,
      set: (s) => {
        v("update:modelValue", s), u.value = s;
      }
    });
    return (s, l) => (c(), f("span", {
      class: r(
        e(g)(
          "relative flex flex-col rounded-xl",
          {
            "focus-within:outline focus-within:outline-offset": e(w)
          },
          t.wrapperClassName
        )
      ),
      "data-testid": "select"
    }, [
      B(N("select", k({
        "onUpdate:modelValue": l[0] || (l[0] = (a) => m.value = a),
        required: t.required,
        disabled: t.disabled,
        class: e(g)(
          "appearance-none disabled:cursor-not-allowed cursor-pointer px-3 text-neutral-900 focus:ring-primary-700 focus:ring-2 outline-hidden bg-transparent rounded-xl ring-1 ring-inset ring-neutral-300 hover:ring-primary-700 active:ring-2 active:ring-primary-700 disabled:bg-disabled-100 disabled:text-disabled-900 disabled:ring-disabled-200",
          {
            "py-1.5": t.size === e(i).sm,
            "py-2": t.size === e(i).base,
            "py-3 text-base": t.size === e(i).lg,
            "!ring-negative-700 ring-2": t.invalid && !t.disabled
          },
          s.$attrs.class
        ),
        "data-testid": "select-input",
        onBlur: l[1] || (l[1] = //@ts-ignore
        (...a) => e(n) && e(n)(...a)),
        onChange: l[2] || (l[2] = //@ts-ignore
        (...a) => e(n) && e(n)(...a)),
        onClick: l[3] || (l[3] = //@ts-ignore
        (...a) => e(o) && e(o)(...a)),
        onKeydown: l[4] || (l[4] = q(
          //@ts-ignore
          (...a) => e(o) && e(o)(...a),
          ["space"]
        ))
      }, e(S)), [
        t.placeholder ? (c(), f("option", {
          key: 0,
          hidden: "",
          class: r(["text-sm bg-neutral-300", [
            "bg-neutral-300 text-sm",
            {
              "text-base": t.size === e(i).lg
            }
          ]]),
          value: "",
          "data-testid": "select-placeholder"
        }, M(t.placeholder), 3)) : $("", !0),
        p(s.$slots, "default")
      ], 16, O), [
        [D, m.value]
      ]),
      p(s.$slots, "chevron", {}, () => [
        E(e(K), {
          class: r([
            "absolute -translate-y-1 pointer-events-none top-1/3 right-4 transition easy-in-out duration-0.5",
            t.disabled ? "text-disabled-500" : "text-neutral-500",
            e(x) ? "rotate-180" : ""
          ])
        }, null, 8, ["class"])
      ])
    ], 2));
  }
});
export {
  H as default
};
