// @polar-sh/sdk/dist/esm/models/operations/customerportaldownloadableslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceDownloadableRead$inboundSchema, } from "../components/listresourcedownloadableread.js";
/** @internal */
export const CustomerPortalDownloadablesListSecurity$outboundSchema = z.pipe(z.object({
    customerSession: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
}));
export function customerPortalDownloadablesListSecurityToJSON(customerPortalDownloadablesListSecurity) {
    return JSON.stringify(CustomerPortalDownloadablesListSecurity$outboundSchema.parse(customerPortalDownloadablesListSecurity));
}
/** @internal */
export const CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function customerPortalDownloadablesListQueryParamBenefitIDFilterToJSON(customerPortalDownloadablesListQueryParamBenefitIDFilter) {
    return JSON.stringify(CustomerPortalDownloadablesListQueryParamBenefitIDFilter$outboundSchema
        .parse(customerPortalDownloadablesListQueryParamBenefitIDFilter));
}
/** @internal */
export const CustomerPortalDownloadablesListRequest$outboundSchema = z.pipe(z.object({
    benefitId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
}), z.transform((v) => {
    return remap$(v, {
        benefitId: "benefit_id",
    });
}));
export function customerPortalDownloadablesListRequestToJSON(customerPortalDownloadablesListRequest) {
    return JSON.stringify(CustomerPortalDownloadablesListRequest$outboundSchema.parse(customerPortalDownloadablesListRequest));
}
/** @internal */
export const CustomerPortalDownloadablesListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceDownloadableRead$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function customerPortalDownloadablesListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalDownloadablesListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalDownloadablesListResponse' from JSON`);
}
