// @polar-sh/sdk/dist/esm/models/operations/refundslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceRefund$inboundSchema, } from "../components/listresourcerefund.js";
import { RefundSortProperty$outboundSchema, } from "../components/refundsortproperty.js";
/** @internal */
export const RefundIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function refundIDFilterToJSON(refundIDFilter) {
    return JSON.stringify(RefundIDFilter$outboundSchema.parse(refundIDFilter));
}
/** @internal */
export const RefundsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function refundsListQueryParamOrganizationIDFilterToJSON(refundsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(RefundsListQueryParamOrganizationIDFilter$outboundSchema.parse(refundsListQueryParamOrganizationIDFilter));
}
/** @internal */
export const OrderIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function orderIDFilterToJSON(orderIDFilter) {
    return JSON.stringify(OrderIDFilter$outboundSchema.parse(orderIDFilter));
}
/** @internal */
export const SubscriptionIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function subscriptionIDFilterToJSON(subscriptionIDFilter) {
    return JSON.stringify(SubscriptionIDFilter$outboundSchema.parse(subscriptionIDFilter));
}
/** @internal */
export const RefundsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function refundsListQueryParamCustomerIDFilterToJSON(refundsListQueryParamCustomerIDFilter) {
    return JSON.stringify(RefundsListQueryParamCustomerIDFilter$outboundSchema.parse(refundsListQueryParamCustomerIDFilter));
}
/** @internal */
export const RefundsListRequest$outboundSchema = z.pipe(z.object({
    id: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    organizationId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    orderId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    subscriptionId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    succeeded: z.optional(z.nullable(z.boolean())),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(RefundSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        orderId: "order_id",
        subscriptionId: "subscription_id",
        customerId: "customer_id",
    });
}));
export function refundsListRequestToJSON(refundsListRequest) {
    return JSON.stringify(RefundsListRequest$outboundSchema.parse(refundsListRequest));
}
/** @internal */
export const RefundsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceRefund$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function refundsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RefundsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefundsListResponse' from JSON`);
}
