// @polar-sh/sdk/dist/esm/models/operations/walletslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceWallet$inboundSchema, } from "../components/listresourcewallet.js";
import { WalletSortProperty$outboundSchema, } from "../components/walletsortproperty.js";
/** @internal */
export const WalletsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function walletsListQueryParamOrganizationIDFilterToJSON(walletsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(WalletsListQueryParamOrganizationIDFilter$outboundSchema.parse(walletsListQueryParamOrganizationIDFilter));
}
/** @internal */
export const WalletsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function walletsListQueryParamCustomerIDFilterToJSON(walletsListQueryParamCustomerIDFilter) {
    return JSON.stringify(WalletsListQueryParamCustomerIDFilter$outboundSchema.parse(walletsListQueryParamCustomerIDFilter));
}
/** @internal */
export const WalletsListRequest$outboundSchema = z.pipe(z.object({
    organizationId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(WalletSortProperty$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        customerId: "customer_id",
    });
}));
export function walletsListRequestToJSON(walletsListRequest) {
    return JSON.stringify(WalletsListRequest$outboundSchema.parse(walletsListRequest));
}
/** @internal */
export const WalletsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceWallet$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function walletsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WalletsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WalletsListResponse' from JSON`);
}
