// @polar-sh/sdk/dist/esm/models/operations/webhookslistwebhookdeliveries.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ListResourceWebhookDelivery$inboundSchema, } from "../components/listresourcewebhookdelivery.js";
/** @internal */
export const EndpointId$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function endpointIdToJSON(endpointId) {
    return JSON.stringify(EndpointId$outboundSchema.parse(endpointId));
}
/** @internal */
export const WebhooksListWebhookDeliveriesRequest$outboundSchema = z.pipe(z.object({
    endpointId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    startTimestamp: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    endTimestamp: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
}), z.transform((v) => {
    return remap$(v, {
        endpointId: "endpoint_id",
        startTimestamp: "start_timestamp",
        endTimestamp: "end_timestamp",
    });
}));
export function webhooksListWebhookDeliveriesRequestToJSON(webhooksListWebhookDeliveriesRequest) {
    return JSON.stringify(WebhooksListWebhookDeliveriesRequest$outboundSchema.parse(webhooksListWebhookDeliveriesRequest));
}
/** @internal */
export const WebhooksListWebhookDeliveriesResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceWebhookDelivery$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function webhooksListWebhookDeliveriesResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebhooksListWebhookDeliveriesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhooksListWebhookDeliveriesResponse' from JSON`);
}
