// @storefront-ui/vue/dist/components/SfRating/SfRating.vue.mjs
import { defineComponent as w, toRefs as N, computed as e, openBlock as t, createElementBlock as o, mergeProps as $, unref as l, Fragment as v, renderList as g, createBlock as u, createCommentVNode as S } from "vue";
import { SfRatingSize as a, clamp as z, roundToNearest as B } from "@storefront-ui/shared";
import { useTwMergeRoot as C, twMerge as R } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import I from "../SfIcons/SfIconStarFilled.vue.mjs";
import L from "../SfIcons/SfIconStarHalf.vue.mjs";
import M from "../SfIcons/SfIconStar.vue.mjs";
const F = ["ariaLabel", "title"], T = {
  [a.xs]: "text-xs",
  [a.sm]: "text-sm",
  [a.base]: "text-base",
  [a.lg]: "text-lg",
  [a.xl]: "text-xl"
}, D = /* @__PURE__ */ w({
  inheritAttrs: !1,
  __name: "SfRating",
  props: {
    size: {
      type: String,
      default: a.base
    },
    max: {
      type: Number,
      default: 5
    },
    value: {
      type: Number,
      default: 0
    },
    halfIncrement: {
      type: Boolean,
      default: !1
    },
    ariaLabel: {
      type: String,
      default: void 0
    }
  },
  setup(m) {
    const x = m, { value: c, max: r, halfIncrement: h, ariaLabel: s } = N(x), { attrsWithoutClass: b } = C(), _ = e(() => h.value ? 0.5 : 1), f = e(() => z(B(c.value, _.value), 0, r.value)), n = e(() => +!Number.isInteger(f.value)), d = e(() => Math.ceil(f.value - n.value)), y = e(() => r.value - d.value - n.value), p = e(() => (s == null ? void 0 : s.value) ?? `${c.value} out of ${r.value}`);
    return (k, V) => (t(), o("span", $({
      role: "img",
      ariaLabel: p.value,
      title: p.value,
      class: l(R)("inline-flex items-center text-warning-500", T[m.size], k.$attrs.class),
      "data-testid": "rating"
    }, l(b)), [
      (t(!0), o(v, null, g(d.value, (i) => (t(), u(l(I), {
        key: i,
        "aria-hidden": "true",
        class: "w-[1.5em] h-[1.5em]"
      }))), 128)),
      n.value ? (t(), u(l(L), {
        key: 0,
        "aria-hidden": "true",
        class: "w-[1.5em] h-[1.5em]"
      })) : S("", !0),
      (t(!0), o(v, null, g(y.value, (i) => (t(), u(l(M), {
        key: i,
        "aria-hidden": "true",
        class: "text-disabled-500 w-[1.5em] h-[1.5em]"
      }))), 128))
    ], 16, F));
  }
});
export {
  D as default
};
