// @grapesjs/studio-sdk-plugins/dist/canvasEmptyState/index.es.js
const U = "app.grapesjs.com", M = "app-stage.grapesjs.com", W = "app2.grapesjs.com", D = "app-stage2.grapesjs.com", I = [
  U,
  W,
  M,
  D,
  "localhost",
  "127.0.0.1",
  ".local-credentialless.webcontainer.io",
  // For stackblitz.com demos
  ".local.webcontainer.io",
  // For stackblitz.com demos
  "-sandpack.codesandbox.io"
  // For Sandpack demos
], P = "license:check:start", _ = "license:check:end", $ = () => typeof window < "u", O = ({ isDev: e, isStage: t }) => `${e ? "" : `https://${t ? M : U}`}/api`, R = () => {
  const e = $() && window.location.hostname;
  return !!e && (I.includes(e) || I.some((t) => e.endsWith(t)));
};
function j(e) {
  return typeof e == "function";
}
async function K({
  path: e,
  baseApiUrl: t,
  method: a = "GET",
  headers: o = {},
  params: c,
  body: u
}) {
  const g = `${t || O({ isDev: !1, isStage: !1 })}${e}`, f = {
    method: a,
    headers: {
      "Content-Type": "application/json",
      ...o
    }
  };
  u && (f.body = JSON.stringify(u));
  const d = c ? new URLSearchParams(c).toString() : "", s = d ? `?${d}` : "", i = await fetch(`${g}${s}`, f);
  if (!i.ok)
    throw new Error(`HTTP error! status: ${i.status}`);
  return i.json();
}
var w = /* @__PURE__ */ ((e) => (e.free = "free", e.startup = "startup", e.business = "business", e.enterprise = "enterprise", e))(w || {});
const N = {
  [w.free]: 0,
  [w.startup]: 10,
  [w.business]: 20,
  [w.enterprise]: 30
};
function G(e) {
  const t = e;
  return t.init = (a) => (o) => e(o, a), t;
}
const x = (e) => /* @__PURE__ */ G(e);
async function F({
  editor: e,
  plan: t,
  pluginName: a,
  licenseKey: o,
  cleanup: c
}) {
  let u = "", h = !1;
  const g = R(), f = (s) => {
    console.warn("Cleanup plugin:", a, "Reason:", s), c();
  }, d = (s = {}) => {
    var C;
    const { error: i, sdkLicense: E } = s, y = (C = s.plan) == null ? void 0 : C.category;
    if (!(E || s.license) || i)
      f(i || "Invalid license");
    else if (y) {
      const L = N[t], T = N[y];
      L > T && f({ pluginRequiredPlan: t, licensePlan: y });
    }
  };
  e.on(P, (s) => {
    u = s == null ? void 0 : s.baseApiUrl, h = !0;
  }), e.on(_, (s) => {
    d(s);
  }), setTimeout(async () => {
    if (!h) {
      if (g) return;
      if (o) {
        const s = await H({ licenseKey: o, pluginName: a, baseApiUrl: u });
        s && d(s);
      } else
        f("The `licenseKey` option not provided");
    }
  }, 2e3);
}
async function H(e) {
  const { licenseKey: t, pluginName: a, baseApiUrl: o } = e;
  try {
    return (await K({
      baseApiUrl: o,
      path: `/sdk/${t || "na"}`,
      method: "POST",
      params: {
        d: window.location.hostname,
        pn: a
      }
    })).result || {};
  } catch (c) {
    return console.error("Error during SDK license check:", c), !1;
  }
}
const q = "canvasEmptyState", B = w.startup, J = function(e, t = {}) {
  const a = /* @__PURE__ */ new WeakMap(), o = /* @__PURE__ */ new WeakMap(), c = /* @__PURE__ */ new WeakMap(), u = /* @__PURE__ */ new Set(), h = /* @__PURE__ */ new WeakMap(), g = {
    emptyStates: [],
    ...t
  }, f = (n, r) => {
    let m = !1;
    const { isValid: p } = r;
    return Array.isArray(p) ? m = p.some((l) => n.is(l)) : j(p) ? m = p({ component: n, editor: e }) : m = n.is(p), m;
  }, d = (n) => {
    const r = a.get(n);
    a.delete(n), r == null || r();
  }, s = (n) => {
    n.views.forEach((r) => d(r)), c.delete(n);
  }, i = (n) => {
    if (!(!n || u.has(n)))
      try {
        u.add(n);
        const r = n.components().length > 0, m = c.get(n);
        if (r && m)
          s(n);
        else if (!r && !m) {
          const p = h.has(n) ? h.get(n) : g.emptyStates.find((l) => f(n, l));
          if (h.set(n, p), !p) return;
          n.views.forEach((l) => {
            const b = p.render({
              editor: e,
              component: n,
              componentView: l,
              mount: (v) => {
                o.set(l, v), c.set(n, !0);
                const A = l.getChildrenContainer();
                A == null || A.appendChild(v);
              },
              unmount: () => d(l)
            });
            a.set(l, () => {
              b == null || b();
              const v = o.get(l);
              v == null || v.remove();
            });
          });
        }
      } finally {
        u.delete(n);
      }
  }, E = (n) => {
    c.has(n) && s(n);
  }, y = ({ model: n }) => {
    i(n.getComponent());
  }, S = e.Components.events, C = `${S.update}:components`, L = "component:mount", T = e.Canvas.events.frameLoadBody, k = S.removed;
  e.on(C, i), e.on(L, i), e.on(k, E), e.on(T, y), F({
    editor: e,
    licenseKey: g.licenseKey,
    plan: B,
    pluginName: q,
    cleanup: () => {
      e.off(C, i), e.off(L, i), e.off(k, E), e.off(T, y);
    }
  });
}, z = x(J);
export {
  z as default
};
