// @polar-sh/sdk/dist/esm/models/operations/customerseatsrevokeseat.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSeatsRevokeSeatRequest$inboundSchema = z.object({
    seat_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "seat_id": "seatId",
    });
});
/** @internal */
export const CustomerSeatsRevokeSeatRequest$outboundSchema = z.object({
    seatId: z.string(),
}).transform((v) => {
    return remap$(v, {
        seatId: "seat_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSeatsRevokeSeatRequest$;
(function (CustomerSeatsRevokeSeatRequest$) {
    /** @deprecated use `CustomerSeatsRevokeSeatRequest$inboundSchema` instead. */
    CustomerSeatsRevokeSeatRequest$.inboundSchema = CustomerSeatsRevokeSeatRequest$inboundSchema;
    /** @deprecated use `CustomerSeatsRevokeSeatRequest$outboundSchema` instead. */
    CustomerSeatsRevokeSeatRequest$.outboundSchema = CustomerSeatsRevokeSeatRequest$outboundSchema;
})(CustomerSeatsRevokeSeatRequest$ || (CustomerSeatsRevokeSeatRequest$ = {}));
export function customerSeatsRevokeSeatRequestToJSON(customerSeatsRevokeSeatRequest) {
    return JSON.stringify(CustomerSeatsRevokeSeatRequest$outboundSchema.parse(customerSeatsRevokeSeatRequest));
}
export function customerSeatsRevokeSeatRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSeatsRevokeSeatRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeatsRevokeSeatRequest' from JSON`);
}
