// @polar-sh/sdk/dist/esm/models/operations/customerseatslistseats.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSeatsListSeatsRequest$inboundSchema = z.object({
    subscription_id: z.nullable(z.string()).optional(),
    order_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
    });
});
/** @internal */
export const CustomerSeatsListSeatsRequest$outboundSchema = z.object({
    subscriptionId: z.nullable(z.string()).optional(),
    orderId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        subscriptionId: "subscription_id",
        orderId: "order_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSeatsListSeatsRequest$;
(function (CustomerSeatsListSeatsRequest$) {
    /** @deprecated use `CustomerSeatsListSeatsRequest$inboundSchema` instead. */
    CustomerSeatsListSeatsRequest$.inboundSchema = CustomerSeatsListSeatsRequest$inboundSchema;
    /** @deprecated use `CustomerSeatsListSeatsRequest$outboundSchema` instead. */
    CustomerSeatsListSeatsRequest$.outboundSchema = CustomerSeatsListSeatsRequest$outboundSchema;
})(CustomerSeatsListSeatsRequest$ || (CustomerSeatsListSeatsRequest$ = {}));
export function customerSeatsListSeatsRequestToJSON(customerSeatsListSeatsRequest) {
    return JSON.stringify(CustomerSeatsListSeatsRequest$outboundSchema.parse(customerSeatsListSeatsRequest));
}
export function customerSeatsListSeatsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSeatsListSeatsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeatsListSeatsRequest' from JSON`);
}
