// @polar-sh/sdk/dist/esm/models/operations/customerseatsresendinvitation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSeatsResendInvitationRequest$inboundSchema = z.object({
    seat_id: z.string(),
}).transform((v) => {
    return remap$(v, {
        "seat_id": "seatId",
    });
});
/** @internal */
export const CustomerSeatsResendInvitationRequest$outboundSchema = z.object({
    seatId: z.string(),
}).transform((v) => {
    return remap$(v, {
        seatId: "seat_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSeatsResendInvitationRequest$;
(function (CustomerSeatsResendInvitationRequest$) {
    /** @deprecated use `CustomerSeatsResendInvitationRequest$inboundSchema` instead. */
    CustomerSeatsResendInvitationRequest$.inboundSchema = CustomerSeatsResendInvitationRequest$inboundSchema;
    /** @deprecated use `CustomerSeatsResendInvitationRequest$outboundSchema` instead. */
    CustomerSeatsResendInvitationRequest$.outboundSchema = CustomerSeatsResendInvitationRequest$outboundSchema;
})(CustomerSeatsResendInvitationRequest$ || (CustomerSeatsResendInvitationRequest$ = {}));
export function customerSeatsResendInvitationRequestToJSON(customerSeatsResendInvitationRequest) {
    return JSON.stringify(CustomerSeatsResendInvitationRequest$outboundSchema.parse(customerSeatsResendInvitationRequest));
}
export function customerSeatsResendInvitationRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSeatsResendInvitationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeatsResendInvitationRequest' from JSON`);
}
