// @polar-sh/sdk/dist/esm/models/operations/customerseatsgetclaiminfo.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerSeatsGetClaimInfoRequest$inboundSchema = z.object({
    invitation_token: z.string(),
}).transform((v) => {
    return remap$(v, {
        "invitation_token": "invitationToken",
    });
});
/** @internal */
export const CustomerSeatsGetClaimInfoRequest$outboundSchema = z.object({
    invitationToken: z.string(),
}).transform((v) => {
    return remap$(v, {
        invitationToken: "invitation_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSeatsGetClaimInfoRequest$;
(function (CustomerSeatsGetClaimInfoRequest$) {
    /** @deprecated use `CustomerSeatsGetClaimInfoRequest$inboundSchema` instead. */
    CustomerSeatsGetClaimInfoRequest$.inboundSchema = CustomerSeatsGetClaimInfoRequest$inboundSchema;
    /** @deprecated use `CustomerSeatsGetClaimInfoRequest$outboundSchema` instead. */
    CustomerSeatsGetClaimInfoRequest$.outboundSchema = CustomerSeatsGetClaimInfoRequest$outboundSchema;
})(CustomerSeatsGetClaimInfoRequest$ || (CustomerSeatsGetClaimInfoRequest$ = {}));
export function customerSeatsGetClaimInfoRequestToJSON(customerSeatsGetClaimInfoRequest) {
    return JSON.stringify(CustomerSeatsGetClaimInfoRequest$outboundSchema.parse(customerSeatsGetClaimInfoRequest));
}
export function customerSeatsGetClaimInfoRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSeatsGetClaimInfoRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeatsGetClaimInfoRequest' from JSON`);
}
