// @polar-sh/sdk/dist/esm/models/operations/ordersexport.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const OrdersExportQueryParamOrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const OrdersExportQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrdersExportQueryParamOrganizationIDFilter$;
(function (OrdersExportQueryParamOrganizationIDFilter$) {
    /** @deprecated use `OrdersExportQueryParamOrganizationIDFilter$inboundSchema` instead. */
    OrdersExportQueryParamOrganizationIDFilter$.inboundSchema = OrdersExportQueryParamOrganizationIDFilter$inboundSchema;
    /** @deprecated use `OrdersExportQueryParamOrganizationIDFilter$outboundSchema` instead. */
    OrdersExportQueryParamOrganizationIDFilter$.outboundSchema = OrdersExportQueryParamOrganizationIDFilter$outboundSchema;
})(OrdersExportQueryParamOrganizationIDFilter$ || (OrdersExportQueryParamOrganizationIDFilter$ = {}));
export function ordersExportQueryParamOrganizationIDFilterToJSON(ordersExportQueryParamOrganizationIDFilter) {
    return JSON.stringify(OrdersExportQueryParamOrganizationIDFilter$outboundSchema.parse(ordersExportQueryParamOrganizationIDFilter));
}
export function ordersExportQueryParamOrganizationIDFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrdersExportQueryParamOrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportQueryParamOrganizationIDFilter' from JSON`);
}
/** @internal */
export const OrdersExportQueryParamProductIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const OrdersExportQueryParamProductIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrdersExportQueryParamProductIDFilter$;
(function (OrdersExportQueryParamProductIDFilter$) {
    /** @deprecated use `OrdersExportQueryParamProductIDFilter$inboundSchema` instead. */
    OrdersExportQueryParamProductIDFilter$.inboundSchema = OrdersExportQueryParamProductIDFilter$inboundSchema;
    /** @deprecated use `OrdersExportQueryParamProductIDFilter$outboundSchema` instead. */
    OrdersExportQueryParamProductIDFilter$.outboundSchema = OrdersExportQueryParamProductIDFilter$outboundSchema;
})(OrdersExportQueryParamProductIDFilter$ || (OrdersExportQueryParamProductIDFilter$ = {}));
export function ordersExportQueryParamProductIDFilterToJSON(ordersExportQueryParamProductIDFilter) {
    return JSON.stringify(OrdersExportQueryParamProductIDFilter$outboundSchema.parse(ordersExportQueryParamProductIDFilter));
}
export function ordersExportQueryParamProductIDFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrdersExportQueryParamProductIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportQueryParamProductIDFilter' from JSON`);
}
/** @internal */
export const OrdersExportRequest$inboundSchema = z.object({
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    product_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
}).transform((v) => {
    return remap$(v, {
        "organization_id": "organizationId",
        "product_id": "productId",
    });
});
/** @internal */
export const OrdersExportRequest$outboundSchema = z.object({
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    productId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
}).transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var OrdersExportRequest$;
(function (OrdersExportRequest$) {
    /** @deprecated use `OrdersExportRequest$inboundSchema` instead. */
    OrdersExportRequest$.inboundSchema = OrdersExportRequest$inboundSchema;
    /** @deprecated use `OrdersExportRequest$outboundSchema` instead. */
    OrdersExportRequest$.outboundSchema = OrdersExportRequest$outboundSchema;
})(OrdersExportRequest$ || (OrdersExportRequest$ = {}));
export function ordersExportRequestToJSON(ordersExportRequest) {
    return JSON.stringify(OrdersExportRequest$outboundSchema.parse(ordersExportRequest));
}
export function ordersExportRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OrdersExportRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrdersExportRequest' from JSON`);
}
