// @storefront-ui/vue/dist/components/SfModal/SfModal.vue.mjs
import { defineComponent as c, toRefs as m, ref as p, unref as t, openBlock as y, createBlock as b, resolveDynamicComponent as w, mergeProps as h, withKeys as C, withCtx as g, renderSlot as k, createCommentVNode as V } from "vue";
import { onClickOutside as v } from "@vueuse/core";
import { useTwMergeRoot as x, twMerge as B } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { useTrapFocus as F } from "../../composables/useTrapFocus/useTrapFocus.mjs";
const K = /* @__PURE__ */ c({
  inheritAttrs: !1,
  __name: "SfModal",
  props: {
    modelValue: {
      type: Boolean,
      default: !1
    },
    tag: {
      type: [String, Object],
      default: ""
    },
    disableClickAway: {
      type: Boolean,
      default: !1
    },
    disableEsc: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(a, { emit: r }) {
    const i = a, { disableClickAway: n, disableEsc: d, modelValue: o } = m(i), l = r, { attrsWithoutClass: u } = x(), e = p();
    v(e, () => {
      n.value || l("update:modelValue", !1);
    });
    const f = () => {
      d.value || l("update:modelValue", !1);
    };
    return F(e, {
      trapTabs: !0,
      activeState: o,
      initialFocus: !1,
      initialFocusContainerFallback: !0
    }), (s, R) => t(o) ? (y(), b(w(a.tag || "div"), h({
      key: 0,
      ref_key: "modalRef",
      ref: e,
      "aria-modal": "true",
      "data-testid": "modal",
      tabindex: "-1",
      class: t(B)(
        "fixed inset-0 w-fit h-fit m-auto p-6 pt-10 lg:p-10 border border-neutral-100 bg-white shadow-xl rounded-3xl outline-hidden",
        s.$attrs.class
      )
    }, t(u), {
      onKeydown: C(f, ["esc"])
    }), {
      default: g(() => [
        k(s.$slots, "default")
      ]),
      _: 3
    }, 16, ["class"])) : V("", !0);
  }
});
export {
  K as default
};
