// @storefront-ui/vue/dist/components/SfProgressCircular/SfProgressCircular.vue.mjs
import { defineComponent as c, toRefs as u, computed as g, openBlock as m, createElementBlock as x, mergeProps as p, unref as s, createElementVNode as d, normalizeClass as f, renderSlot as h } from "vue";
import { useTwMergeRoot as k, twMerge as o } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfProgressSize as e } from "@storefront-ui/shared";
import { ClassProp as w } from "../../shared/props.mjs";
const b = ["aria-valuenow", "aria-label", "stroke-dasharray"], v = {
  [e.xs]: "h-4 w-4 ring-2",
  [e.sm]: "h-5 w-5 ring-2",
  [e.base]: "h-6 w-6 ring-2",
  [e.lg]: "h-8 w-8 ring-2",
  [e.xl]: "h-10 w-10 ring-2",
  [e["2xl"]]: "h-14 w-14 ring-[3px]",
  [e["3xl"]]: "h-24 w-24 ring-4",
  [e["4xl"]]: "h-48 w-48 ring-8"
}, y = {
  [e.xs]: "stroke-[10px]",
  [e.sm]: "stroke-[8px]",
  [e.base]: "stroke-[6px]",
  [e.lg]: "stroke-[4px]",
  [e.xl]: "stroke-[3px]",
  [e["2xl"]]: "stroke-[3px]",
  [e["3xl"]]: "stroke-2",
  [e["4xl"]]: "stroke-2"
}, $ = /* @__PURE__ */ c({
  inheritAttrs: !1,
  __name: "SfProgressCircular",
  props: {
    value: {
      type: Number,
      default: 0
    },
    size: {
      type: String,
      default: e.base
    },
    ariaLabel: {
      type: String,
      default: "Progress element"
    },
    circleClass: w
  },
  setup(r) {
    const l = r, { value: t } = u(l), { attrsWithoutClass: i } = k(), n = g(() => `${t.value / 100 * 151}, 150`);
    return (a, C) => (m(), x("svg", p({
      role: "progressbar",
      "aria-valuemin": "0",
      "aria-valuemax": "100",
      "aria-valuenow": s(t),
      "aria-label": r.ariaLabel,
      class: s(o)(
        "inline-block ring-inset ring-neutral-300 text-primary-700 rounded-full transition-[stroke-dasharray] ease-in-out duration-500 text-sm",
        v[r.size],
        a.$attrs.class
      ),
      viewBox: "25 25 50 50",
      "stroke-dasharray": n.value,
      "data-testid": "progress"
    }, s(i)), [
      d("circle", {
        class: f(s(o)("origin-bottom-right -rotate-90 stroke-current fill-none", y[r.size], r.circleClass)),
        cx: "50",
        cy: "50",
        r: "24"
      }, null, 2),
      h(a.$slots, "default")
    ], 16, b));
  }
});
export {
  $ as default
};
