// @storefront-ui/vue/dist/components/SfLoaderCircular/SfLoaderCircular.vue.mjs
import { defineComponent as i, openBlock as o, createElementBlock as n, mergeProps as c, unref as s, createElementVNode as x, normalizeClass as m } from "vue";
import { useTwMergeRoot as g, twMerge as t } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfLoaderSize as e } from "@storefront-ui/shared";
import { ClassProp as p } from "../../shared/props.mjs";
const u = ["aria-label"], d = {
  [e.xs]: "h-4 w-4 ring-2",
  [e.sm]: "h-5 w-5 ring-2",
  [e.base]: "h-6 w-6 ring-2",
  [e.lg]: "h-8 w-8 ring-2",
  [e.xl]: "h-10 w-10 ring-2",
  [e["2xl"]]: "h-14 w-14 ring-[3px]",
  [e["3xl"]]: "h-24 w-24 ring-4",
  [e["4xl"]]: "h-48 w-48 ring-8"
}, f = {
  [e.xs]: "stroke-[10px]",
  [e.sm]: "stroke-[8px]",
  [e.base]: "stroke-[6px]",
  [e.lg]: "stroke-[4px]",
  [e.xl]: "stroke-[3px]",
  [e["2xl"]]: "stroke-[3px]",
  [e["3xl"]]: "stroke-2",
  [e["4xl"]]: "stroke-2"
}, z = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "SfLoaderCircular",
  props: {
    size: {
      type: String,
      default: e.base
    },
    ariaLabel: {
      type: String,
      default: "loading"
    },
    circleClass: p
  },
  setup(r) {
    const { attrsWithoutClass: a } = g();
    return (l, k) => (o(), n("svg", c({
      class: s(t)(
        "inline-block rounded-full ring-inset ring-neutral-300 text-primary-700 animate-spin-slow",
        d[r.size],
        l.$attrs.class
      ),
      viewBox: "25 25 50 50",
      "aria-live": "polite",
      "aria-label": r.ariaLabel,
      "data-testid": "loader-circular"
    }, s(a)), [
      x("circle", {
        class: m(
          s(t)("stroke-current stroke-2 fill-none animate-stroke-loader-circular", f[r.size], r.circleClass)
        ),
        cx: "50",
        cy: "50",
        r: "24"
      }, null, 2)
    ], 16, u));
  }
});
export {
  z as default
};
