// @storefront-ui/vue/dist/components/SfCheckbox/SfCheckbox.vue.mjs
import { defineComponent as v, mergeModels as p, mergeDefaults as h, useModel as k, useTemplateRef as b, ref as m, watch as x, openBlock as l, createBlock as r, resolveDynamicComponent as y, normalizeClass as g, unref as a, withCtx as w, withDirectives as C, createElementVNode as $, mergeProps as M, vModelCheckbox as B, createElementBlock as R, Fragment as V } from "vue";
import { useTwMergeRoot as T, twMerge as c } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import D from "../SfIcons/SfIconIndeterminateCheckBox.vue.mjs";
import E from "../SfIcons/SfIconCheckBox.vue.mjs";
import P from "../SfIcons/SfIconCheckBoxOutlineBlank.vue.mjs";
import { ClassProp as S } from "../../shared/props.mjs";
const z = ["indeterminate", "invalid"], j = /* @__PURE__ */ v({
  inheritAttrs: !1,
  __name: "SfCheckbox",
  props: /* @__PURE__ */ p(/* @__PURE__ */ h({
    invalid: {
      type: Boolean,
      default: !1
    },
    indeterminate: {
      type: Boolean,
      default: !1
    },
    wrapperTag: {
      type: String,
      default: "label"
    },
    wrapperClass: S
  }, {
    indeterminate: !1
  }), {
    modelValue: { default: !1 },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(e) {
    const { attrsWithoutClass: f } = T(), n = k(e, "modelValue"), o = b("checkboxRef"), s = m(), d = m();
    return x(
      [n, () => e.indeterminate],
      async () => {
        var t, i;
        s.value = (t = o.value) == null ? void 0 : t.checked, d.value = (i = o.value) == null ? void 0 : i.indeterminate;
      },
      { immediate: !0, flush: "post" }
    ), (t, i) => (l(), r(y(e.wrapperTag), {
      class: g(
        a(c)(
          "flex cursor-pointer focus-visible-within:outline focus-visible-within:outline-offset-2 rounded-md",
          {
            "text-neutral-500 hover:text-primary-800 active:text-primary-900": !e.invalid && !t.$attrs.disabled,
            "text-negative-700 hover:text-negative-800 active:text-negative-900": e.invalid && !t.$attrs.disabled,
            "text-disabled-500 hover:text-disabled-600 active:text-disabled-700": t.$attrs.disabled
          },
          e.wrapperClass
        )
      ),
      "data-testid": "checkbox"
    }, {
      default: w(() => [
        C($("input", M({
          class: a(c)("sr-only", t.$attrs.class),
          type: "checkbox",
          ref_key: "checkboxRef",
          ref: o,
          indeterminate: e.indeterminate,
          invalid: e.invalid
        }, a(f), {
          "onUpdate:modelValue": i[0] || (i[0] = (u) => n.value = u)
        }), null, 16, z), [
          [B, n.value]
        ]),
        d.value ? (l(), r(a(D), { key: 0 })) : (l(), R(V, { key: 1 }, [
          s.value ? (l(), r(a(E), { key: 0 })) : (l(), r(a(P), { key: 1 }))
        ], 64))
      ]),
      _: 1
    }, 8, ["class"]));
  }
});
export {
  j as default
};
