// @polar-sh/sdk/dist/esm/models/components/customerseatclaimresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomerSeat$inboundSchema, CustomerSeat$outboundSchema, } from "./customerseat.js";
/** @internal */
export const CustomerSeatClaimResponse$inboundSchema = z.object({
    seat: CustomerSeat$inboundSchema,
    customer_session_token: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session_token": "customerSessionToken",
    });
});
/** @internal */
export const CustomerSeatClaimResponse$outboundSchema = z.object({
    seat: CustomerSeat$outboundSchema,
    customerSessionToken: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSessionToken: "customer_session_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerSeatClaimResponse$;
(function (CustomerSeatClaimResponse$) {
    /** @deprecated use `CustomerSeatClaimResponse$inboundSchema` instead. */
    CustomerSeatClaimResponse$.inboundSchema = CustomerSeatClaimResponse$inboundSchema;
    /** @deprecated use `CustomerSeatClaimResponse$outboundSchema` instead. */
    CustomerSeatClaimResponse$.outboundSchema = CustomerSeatClaimResponse$outboundSchema;
})(CustomerSeatClaimResponse$ || (CustomerSeatClaimResponse$ = {}));
export function customerSeatClaimResponseToJSON(customerSeatClaimResponse) {
    return JSON.stringify(CustomerSeatClaimResponse$outboundSchema.parse(customerSeatClaimResponse));
}
export function customerSeatClaimResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerSeatClaimResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerSeatClaimResponse' from JSON`);
}
