// @polar-sh/sdk/dist/esm/models/components/seatclaim.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SeatClaim$inboundSchema = z.object({
    invitation_token: z.string(),
}).transform((v) => {
    return remap$(v, {
        "invitation_token": "invitationToken",
    });
});
/** @internal */
export const SeatClaim$outboundSchema = z.object({
    invitationToken: z.string(),
}).transform((v) => {
    return remap$(v, {
        invitationToken: "invitation_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SeatClaim$;
(function (SeatClaim$) {
    /** @deprecated use `SeatClaim$inboundSchema` instead. */
    SeatClaim$.inboundSchema = SeatClaim$inboundSchema;
    /** @deprecated use `SeatClaim$outboundSchema` instead. */
    SeatClaim$.outboundSchema = SeatClaim$outboundSchema;
})(SeatClaim$ || (SeatClaim$ = {}));
export function seatClaimToJSON(seatClaim) {
    return JSON.stringify(SeatClaim$outboundSchema.parse(seatClaim));
}
export function seatClaimFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SeatClaim$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SeatClaim' from JSON`);
}
