// @polar-sh/sdk/dist/esm/models/components/seatclaiminfo.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const SeatClaimInfo$inboundSchema = z.object({
    product_name: z.string(),
    product_id: z.string(),
    organization_name: z.string(),
    organization_slug: z.string(),
    customer_email: z.string(),
    can_claim: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        "product_name": "productName",
        "product_id": "productId",
        "organization_name": "organizationName",
        "organization_slug": "organizationSlug",
        "customer_email": "customerEmail",
        "can_claim": "canClaim",
    });
});
/** @internal */
export const SeatClaimInfo$outboundSchema = z.object({
    productName: z.string(),
    productId: z.string(),
    organizationName: z.string(),
    organizationSlug: z.string(),
    customerEmail: z.string(),
    canClaim: z.boolean(),
}).transform((v) => {
    return remap$(v, {
        productName: "product_name",
        productId: "product_id",
        organizationName: "organization_name",
        organizationSlug: "organization_slug",
        customerEmail: "customer_email",
        canClaim: "can_claim",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var SeatClaimInfo$;
(function (SeatClaimInfo$) {
    /** @deprecated use `SeatClaimInfo$inboundSchema` instead. */
    SeatClaimInfo$.inboundSchema = SeatClaimInfo$inboundSchema;
    /** @deprecated use `SeatClaimInfo$outboundSchema` instead. */
    SeatClaimInfo$.outboundSchema = SeatClaimInfo$outboundSchema;
})(SeatClaimInfo$ || (SeatClaimInfo$ = {}));
export function seatClaimInfoToJSON(seatClaimInfo) {
    return JSON.stringify(SeatClaimInfo$outboundSchema.parse(seatClaimInfo));
}
export function seatClaimInfoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SeatClaimInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SeatClaimInfo' from JSON`);
}
