// @nuxt/scripts/dist/runtime/components/GoogleMaps/ScriptGoogleMapsInfoWindow.d.vue.ts
type __VLS_Props = {
    options?: google.maps.InfoWindowOptions;
};
declare var __VLS_1: {};
type __VLS_Slots = {} & {
    default?: (props: typeof __VLS_1) => any;
};
declare const __VLS_base: import("vue").DefineComponent<__VLS_Props, {}, {}, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {} & {
    close: () => any;
    visible: () => any;
    position_changed: () => any;
    zindex_changed: () => any;
    closeclick: () => any;
    content_changed: () => any;
    domready: () => any;
    headercontent_changed: () => any;
    headerdisabled_changed: () => any;
}, string, import("vue").PublicProps, Readonly<__VLS_Props> & Readonly<{
    onClose?: (() => any) | undefined;
    onVisible?: (() => any) | undefined;
    onPosition_changed?: (() => any) | undefined;
    onZindex_changed?: (() => any) | undefined;
    onCloseclick?: (() => any) | undefined;
    onContent_changed?: (() => any) | undefined;
    onDomready?: (() => any) | undefined;
    onHeadercontent_changed?: (() => any) | undefined;
    onHeaderdisabled_changed?: (() => any) | undefined;
}>, {}, {}, {}, {}, string, import("vue").ComponentProvideOptions, false, {}, any>;
declare const __VLS_export: __VLS_WithSlots<typeof __VLS_base, __VLS_Slots>;
declare const _default: typeof __VLS_export;
export default _default;
type __VLS_WithSlots<T, S> = T & {
    new (): {
        $slots: S;
    };
};
