// @storefront-ui/vue/dist/components/SfLoaderLinear/SfLoaderLinear.vue.mjs
import { defineComponent as i, openBlock as o, createElementBlock as s, mergeProps as n, unref as t } from "vue";
import { useTwMergeRoot as f, twMerge as m } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfLoaderLinearSize as c } from "./types.mjs";
import { SfLoaderSize as e } from "@storefront-ui/shared";
const h = ["aria-label"], d = {
  [c.minimal]: "h-1",
  [e.xs]: "h-4",
  [e.sm]: "h-5",
  [e.base]: "h-6",
  [e.lg]: "h-7",
  [e.xl]: "h-10",
  [e["2xl"]]: "h-14",
  [e["3xl"]]: "h-24",
  [e["4xl"]]: "h-48"
}, L = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "SfLoaderLinear",
  props: {
    size: {
      type: String,
      default: e.base
    },
    ariaLabel: {
      type: String,
      default: "loading"
    }
  },
  setup(a) {
    const { attrsWithoutClass: r } = f();
    return (l, p) => (o(), s("span", n({
      class: t(m)(
        "relative inline-block overflow-hidden bg-neutral-300 text-primary-700 after:absolute after:w-2.5 after:h-full after:animate-line after:bg-current after:block",
        d[a.size],
        l.$attrs.class
      ),
      "aria-live": "polite",
      "aria-label": a.ariaLabel,
      "data-testid": "loader-linear"
    }, t(r)), null, 16, h));
  }
});
export {
  L as default
};
