// @storefront-ui/vue/dist/components/SfAccordionItem/SfAccordionItem.vue.mjs
import { defineComponent as l, openBlock as r, createElementBlock as m, createElementVNode as a, mergeProps as i, unref as n, withModifiers as u, renderSlot as s } from "vue";
import { twMerge as d } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { ClassProp as f } from "../../shared/props.mjs";
const c = ["open"], k = /* @__PURE__ */ l({
  __name: "SfAccordionItem",
  props: {
    modelValue: {
      type: Boolean,
      default: !1
    },
    summaryClass: f,
    summaryAttrs: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["update:modelValue"],
  setup(e) {
    return (t, o) => (r(), m("details", {
      open: e.modelValue,
      "data-testid": "accordion-item"
    }, [
      a("summary", i(e.summaryAttrs, {
        class: n(d)(
          e.summaryClass,
          "list-none [&::-webkit-details-marker]:hidden cursor-pointer focus-visible:outline focus-visible:outline-offset focus-visible:rounded-xs"
        ),
        onClick: o[0] || (o[0] = u((p) => t.$emit("update:modelValue", !e.modelValue), ["prevent"]))
      }), [
        s(t.$slots, "summary")
      ], 16),
      s(t.$slots, "default")
    ], 8, c));
  }
});
export {
  k as default
};
