// @storefront-ui/vue/dist/components/SfListItem/SfListItem.vue.mjs
import { defineComponent as b, openBlock as l, createBlock as a, resolveDynamicComponent as i, mergeProps as g, unref as s, withCtx as d, normalizeClass as o, renderSlot as r, createCommentVNode as c } from "vue";
import { useTwMergeRoot as x, twMerge as f } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfListItemSize as n } from "@storefront-ui/shared";
import { ClassProp as u } from "../../shared/props.mjs";
const C = {
  [n.sm]: "text-sm px-4 py-1",
  [n.base]: "px-4 py-2",
  [n.lg]: "p-4"
}, S = /* @__PURE__ */ b({
  inheritAttrs: !1,
  __name: "SfListItem",
  props: {
    size: {
      type: String,
      default: n.base
    },
    disabled: {
      type: Boolean,
      default: void 0
    },
    selected: {
      type: Boolean,
      default: !1
    },
    tag: {
      type: [String, Object],
      default: void 0
    },
    childrenTag: {
      type: String,
      default: "span"
    },
    prefixClass: u,
    suffixClass: u,
    defaultClass: u
  },
  setup(e) {
    const { attrsWithoutClass: m } = x();
    return (t, v) => (l(), a(i(e.tag || "li"), g({
      class: s(f)(
        "inline-flex items-center gap-2 w-full hover:bg-neutral-100 active:bg-neutral-200 cursor-pointer focus-visible:outline focus-visible:outline-offset focus-visible:relative focus-visible:z-10",
        { "cursor-not-allowed pointer-events-none text-disabled-900": e.disabled, "font-medium": e.selected },
        C[e.size],
        t.$attrs.class
      ),
      disabled: e.disabled,
      "data-testid": "list-item"
    }, s(m)), {
      default: d(() => [
        t.$slots.prefix ? (l(), a(i(e.childrenTag), {
          key: 0,
          class: o(s(f)(e.disabled ? "text-disabled-500" : "text-neutral-500", e.prefixClass))
        }, {
          default: d(() => [
            r(t.$slots, "prefix")
          ]),
          _: 3
        }, 8, ["class"])) : c("", !0),
        (l(), a(i(e.childrenTag), {
          class: o(s(f)("flex flex-col w-full min-w-0", e.defaultClass))
        }, {
          default: d(() => [
            r(t.$slots, "default")
          ]),
          _: 3
        }, 8, ["class"])),
        t.$slots.suffix ? (l(), a(i(e.childrenTag), {
          key: 1,
          class: o(s(f)(e.disabled ? "text-disabled-500" : "text-neutral-500", e.suffixClass))
        }, {
          default: d(() => [
            r(t.$slots, "suffix")
          ]),
          _: 3
        }, 8, ["class"])) : c("", !0)
      ]),
      _: 3
    }, 16, ["class", "disabled"]));
  }
});
export {
  S as default,
  C as sizeClasses
};
