// prosemirror-replaceattrs/build/module/lib/ReplaceAttrsStep.js
// Copyright (C) 2020, Orkhan Alikhanov <http://orkhanalikhanov.com>. All rights reserved.
import { Fragment, Slice } from 'prosemirror-model';
import { Step, StepResult } from 'prosemirror-transform';
export class ReplaceAttrsStep extends Step {
    /**
     * @param at The position of node whose attrs should be updated.
     * @param attrs New attribues to set.
     */
    constructor(at, attrs) {
        super();
        this.at = at;
        this.attrs = attrs;
    }
    /**
     * {@inheritDoc}
     */
    static fromJSON(_schema, json) {
        return new ReplaceAttrsStep(json.at, json.attrs);
    }
    /**
     * {@inheritDoc}
     */
    apply(doc) {
        const node = doc.nodeAt(this.at);
        if (!node) {
            return StepResult.fail('Node could not be found');
        }
        const newNode = node.copy();
        newNode.attrs = this.attrs;
        return StepResult.ok(doc.replace(this.at, this.at + 1, new Slice(Fragment.from(newNode), 0, 0)));
    }
    /**
     * {@inheritDoc}
     */
    invert(doc) {
        return new ReplaceAttrsStep(this.at, doc.nodeAt(this.at).attrs);
    }
    /**
     * {@inheritDoc}
     */
    map(mapping) {
        const at = mapping.mapResult(this.at);
        return at.deleted ? null : new ReplaceAttrsStep(this.at, this.attrs);
    }
    /**
     * {@inheritDoc}
     */
    merge(other) {
        if (other instanceof ReplaceAttrsStep && other.at === this.at) {
            return new ReplaceAttrsStep(this.at, this.attrs);
        }
        return null;
    }
    /**
     * {@inheritDoc}
     */
    toJSON() {
        return {
            attrs: this.attrs,
            at: this.at,
            stepType: 'replaceAttrs'
        };
    }
}
Step.jsonID('replaceAttrs', ReplaceAttrsStep);
