// @nuxtjs/mcp-toolkit/dist/runtime/server/mcp/handler.js
import { getRouterParam } from "h3";
import { useRuntimeConfig } from "#imports";
import { tools } from "#nuxt-mcp/tools.mjs";
import { resources } from "#nuxt-mcp/resources.mjs";
import { prompts } from "#nuxt-mcp/prompts.mjs";
import { handlers } from "#nuxt-mcp/handlers.mjs";
import { defaultHandler } from "#nuxt-mcp/default-handler.mjs";
import { createMcpHandler } from "./utils.js";
import { getMcpConfig } from "./config.js";
export default createMcpHandler((event) => {
  const runtimeConfig = useRuntimeConfig(event).mcp;
  const config = getMcpConfig(runtimeConfig);
  const handlerName = getRouterParam(event, "handler");
  if (handlerName) {
    const handlerDef = handlers.find(
      (h) => h.name === handlerName
    );
    if (!handlerDef) {
      throw new Error(`Handler "${handlerName}" not found`);
    }
    return {
      name: handlerDef.name ?? handlerName,
      version: handlerDef.version ?? config.version,
      browserRedirect: handlerDef.browserRedirect ?? config.browserRedirect,
      tools: handlerDef.tools,
      resources: handlerDef.resources,
      prompts: handlerDef.prompts,
      middleware: handlerDef.middleware
    };
  }
  const defaultHandlerDef = defaultHandler;
  if (defaultHandlerDef) {
    return {
      name: defaultHandlerDef.name ?? config.name ?? "MCP Server",
      version: defaultHandlerDef.version ?? config.version,
      browserRedirect: defaultHandlerDef.browserRedirect ?? config.browserRedirect,
      // Use handler's definitions if specified, otherwise use global definitions
      tools: defaultHandlerDef.tools ?? tools,
      resources: defaultHandlerDef.resources ?? resources,
      prompts: defaultHandlerDef.prompts ?? prompts,
      middleware: defaultHandlerDef.middleware
    };
  }
  return {
    name: config.name || "MCP Server",
    version: config.version,
    browserRedirect: config.browserRedirect,
    tools,
    resources,
    prompts
  };
});
