// @storefront-ui/vue/dist/components/SfBadge/SfBadge.vue.mjs
import { defineComponent as i, computed as n, openBlock as m, createElementBlock as p, mergeProps as u, unref as r, toDisplayString as c } from "vue";
import { useTwMergeRoot as f, twMerge as d } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfBadgePlacement as g, SfBadgeVariant as x } from "@storefront-ui/shared";
const S = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "SfBadge",
  props: {
    content: {
      type: [String, Number],
      default: ""
    },
    max: {
      type: Number,
      default: 99
    },
    placement: {
      type: String,
      default: g["top-right"]
    },
    variant: {
      type: String,
      default: x.standard
    }
  },
  setup(e) {
    const t = e, { attrsWithoutClass: o } = f(), a = n(() => t.variant === "dot"), l = n(() => a.value ? "" : !Number.isNaN(t.content) && Number(t.content) > t.max ? `${t.max}+` : t.content);
    return (s, b) => (m(), p("span", u({
      class: r(d)(
        "block absolute py-0.5 px-1 bg-negative-700 font-medium text-white text-[8px] leading-[8px] rounded-xl",
        {
          "min-w-[12px] min-h-[12px]": !a.value,
          "w-[10px] h-[10px]": a.value,
          "top-0 right-0 -translate-x-0.5 translate-y-0.5": e.placement === "top-right",
          "top-0 left-0 translate-x-0.5 translate-y-0.5": e.placement === "top-left",
          "bottom-0 right-0 -translate-x-0.5 -translate-y-0.5": e.placement === "bottom-right",
          "bottom-0 left-0 translate-x-0.5 -translate-y-0.5": e.placement === "bottom-left"
        },
        s.$attrs.class
      ),
      "data-testid": "badge"
    }, r(o)), c(l.value), 17));
  }
});
export {
  S as default
};
