// @storefront-ui/vue/dist/components/SfProgressLinear/SfProgressLinear.vue.mjs
import { defineComponent as i, openBlock as o, createElementBlock as l, mergeProps as n, unref as s } from "vue";
import { useTwMergeRoot as u, twMerge as g } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfProgressLinearSize as m } from "./types.mjs";
import { SfProgressSize as e } from "@storefront-ui/shared";
const p = ["aria-label", "value"], b = {
  [m.minimal]: "h-1",
  [e.xs]: "h-4",
  [e.sm]: "h-5",
  [e.base]: "h-6",
  [e.lg]: "h-7",
  [e.xl]: "h-10",
  [e["2xl"]]: "h-14",
  [e["3xl"]]: "h-24",
  [e["4xl"]]: "h-48"
}, w = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "SfProgressLinear",
  props: {
    value: {
      type: Number,
      default: 0
    },
    size: {
      type: String,
      default: e.base
    },
    ariaLabel: {
      type: String,
      default: "Progress element"
    }
  },
  setup(r) {
    const { attrsWithoutClass: t } = u();
    return (a, f) => (o(), l("progress", n({
      "data-testid": "progress-linear",
      max: "100",
      class: s(g)(
        "bg-neutral-300 text-primary-700 [&::-webkit-progress-bar]:bg-inherit [&::-webkit-progress-value]:bg-current [&::-webkit-progress-value]:transition-[width] [&::-webkit-progress-value]:ease-in-out [&::-webkit-progress-value]:duration-200 [&::-moz-progress-bar]:bg-current",
        b[r.size],
        a.$attrs.class
      ),
      "aria-label": r.ariaLabel,
      value: r.value
    }, s(t)), null, 16, p));
  }
});
export {
  w as default
};
