// @grapesjs/studio-sdk-plugins/dist/presetPrintable/index.es.js
const W = "app.grapesjs.com", C = "app-stage.grapesjs.com", X = "app2.grapesjs.com", Z = "app-stage2.grapesjs.com", U = [
  W,
  X,
  C,
  Z,
  "localhost",
  "127.0.0.1",
  ".local-credentialless.webcontainer.io",
  // For stackblitz.com demos
  ".local.webcontainer.io",
  // For stackblitz.com demos
  "-sandpack.codesandbox.io"
  // For Sandpack demos
], Q = "license:check:start", ee = "license:check:end", te = () => typeof window < "u", O = (e, o = {}) => {
  const t = ne(e) ? { id: e, src: e } : e;
  return new Promise((s, a) => {
    var g, h;
    const { loadedScripts: n } = o, { id: i, src: p } = t, c = document.querySelector(`script[src="${p}"]`);
    if (c) {
      if ((g = o.onScript) == null || g.call(o, c), n && !n.get(p)) {
        c.addEventListener("load", () => s(i)), c.addEventListener("error", () => a(i));
        return;
      }
      return s(i);
    }
    const u = window.define;
    window.define = void 0;
    const l = () => {
      n == null || n.set(p, !0), window.define = u;
    }, r = document.createElement("script");
    r.src = p, r.onload = () => {
      s(i), l();
    }, r.onerror = () => {
      a(i), l();
    }, document.head.appendChild(r), (h = o.onScript) == null || h.call(o, r);
  });
}, se = ({ isDev: e, isStage: o }) => `${e ? "" : `https://${o ? C : W}`}/api`, oe = () => {
  const e = te() && window.location.hostname;
  return !!e && (U.includes(e) || U.some((o) => e.endsWith(o)));
}, ne = (e) => typeof e == "string";
async function ae({
  path: e,
  baseApiUrl: o,
  method: t = "GET",
  headers: s = {},
  params: a,
  body: n
}) {
  const p = `${o || se({ isDev: !1, isStage: !1 })}${e}`, c = {
    method: t,
    headers: {
      "Content-Type": "application/json",
      ...s
    }
  };
  n && (c.body = JSON.stringify(n));
  const u = a ? new URLSearchParams(a).toString() : "", l = u ? `?${u}` : "", r = await fetch(`${p}${l}`, c);
  if (!r.ok)
    throw new Error(`HTTP error! status: ${r.status}`);
  return r.json();
}
var R = /* @__PURE__ */ ((e) => (e.free = "free", e.startup = "startup", e.business = "business", e.enterprise = "enterprise", e))(R || {}), z = /* @__PURE__ */ ((e) => (e.toastAdd = "studio:toastAdd", e.toastRemove = "studio:toastRemove", e.dialogOpen = "studio:dialogOpen", e.dialogClose = "studio:dialogClose", e.dialogExportCode = "studio:dialogExportCode", e.sidebarLeftSet = "studio:sidebarLeft:set", e.sidebarLeftGet = "studio:sidebarLeft:get", e.sidebarLeftToggle = "studio:sidebarLeft:toggle", e.sidebarRightSet = "studio:sidebarRight:set", e.sidebarRightGet = "studio:sidebarRight:get", e.sidebarRightToggle = "studio:sidebarRight:toggle", e.sidebarTopSet = "studio:sidebarTop:set", e.sidebarTopGet = "studio:sidebarTop:get", e.sidebarTopToggle = "studio:sidebarTop:toggle", e.sidebarBottomSet = "studio:sidebarBottom:set", e.sidebarBottomGet = "studio:sidebarBottom:get", e.sidebarBottomToggle = "studio:sidebarBottom:toggle", e.symbolAdd = "studio:symbolAdd", e.symbolDetach = "studio:symbolDetach", e.symbolOverride = "studio:symbolOverride", e.symbolPropagateStyles = "studio:propagateStyles", e.getPagesConfig = "studio:getPagesConfig", e.setPagesConfig = "studio:setPagesConfig", e.getPageSettings = "studio:getPageSettings", e.setPageSettings = "studio:setPageSettings", e.clearPage = "studio:clearPage", e.projectFiles = "studio:projectFiles", e.validateCode = "studio:validateCode", e.canvasReload = "studio:canvasReload", e.getBlocksPanel = "studio:getBlocksPanel", e.setBlocksPanel = "studio:setBlocksPanel", e.getStateContextMenu = "studio:getStateContextMenu", e.setStateContextMenu = "studio:setStateContextMenu", e.contextMenuComponent = "studio:contextMenuComponent", e.layoutAdd = "studio:layoutAdd", e.layoutRemove = "studio:layoutRemove", e.layoutToggle = "studio:layoutToggle", e.layoutUpdate = "studio:layoutUpdate", e.layoutGet = "studio:layoutGet", e.layoutConfigGet = "studio:layoutConfigGet", e.layoutConfigSet = "studio:layoutConfigSet", e.getStateTheme = "studio:getStateTheme", e.setStateTheme = "studio:setStateTheme", e.assetProviderGet = "studio:assetProviderGet", e.assetProviderAdd = "studio:assetProviderAdd", e.assetProviderRemove = "studio:assetProviderRemove", e.fontGet = "studio:fontGet", e.fontAdd = "studio:fontAdd", e.fontRemove = "studio:fontRemove", e.fontManagerOpen = "studio:fontManagerOpen", e.menuFontLoad = "studio:menuFontLoad", e.toggleStateDataSource = "studio:toggleStateDataSource", e.getStateDataSource = "studio:getStateDataSource", e.dataSourceSetGlobalData = "studio:dataSourceSetGlobalData", e.dataSourceSetImporter = "studio:dataSourceSetImporter", e.dataSourceSetExporter = "studio:dataSourceSetExporter", e.setDragAbsolute = "studio:setDragAbsolute", e))(z || {});
const I = {
  [R.free]: 0,
  [R.startup]: 10,
  [R.business]: 20,
  [R.enterprise]: 30
};
function ie(e) {
  const o = e;
  return o.init = (t) => (s) => e(s, t), o;
}
const re = (e) => /* @__PURE__ */ ie(e);
async function ce({
  editor: e,
  plan: o,
  pluginName: t,
  licenseKey: s,
  cleanup: a
}) {
  let n = "", i = !1;
  const p = oe(), c = (l) => {
    console.warn("Cleanup plugin:", t, "Reason:", l), a();
  }, u = (l = {}) => {
    var w;
    const { error: r, sdkLicense: g } = l, h = (w = l.plan) == null ? void 0 : w.category;
    if (!(g || l.license) || r)
      c(r || "Invalid license");
    else if (h) {
      const d = I[o], f = I[h];
      d > f && c({ pluginRequiredPlan: o, licensePlan: h });
    }
  };
  e.on(Q, (l) => {
    n = l == null ? void 0 : l.baseApiUrl, i = !0;
  }), e.on(ee, (l) => {
    u(l);
  }), setTimeout(async () => {
    if (!i) {
      if (p) return;
      if (s) {
        const l = await le({ licenseKey: s, pluginName: t, baseApiUrl: n });
        l && u(l);
      } else
        c("The `licenseKey` option not provided");
    }
  }, 2e3);
}
async function le(e) {
  const { licenseKey: o, pluginName: t, baseApiUrl: s } = e;
  try {
    return (await ae({
      baseApiUrl: s,
      path: `/sdk/${o || "na"}`,
      method: "POST",
      params: {
        d: window.location.hostname,
        pn: t
      }
    })).result || {};
  } catch (a) {
    return console.error("Error during SDK license check:", a), !1;
  }
}
const de = (e) => (o) => {
  var t;
  return ((t = o.getAttribute) == null ? void 0 : t.call(o, V)) === e;
}, pe = (...e) => (o, t) => e.some((s) => t.is(s)), K = (e, o, t) => {
  const { doc: s = document } = t, a = t.target || s.head, n = a.querySelector(`style[${e}]`) || s.createElement("style");
  n.setAttribute(e, "true"), n.innerHTML = o, !a.contains(n) && a.appendChild(n);
}, ge = "gjs-plg-", q = "data-gs-plg-", V = "data-type-role";
function N(e, o, t) {
  let s, a, n, i, p;
  const c = o === void 0 ? 100 : o;
  function u() {
    const r = Date.now() - i;
    r < c && r >= 0 ? s = setTimeout(u, c - r) : (s = null, p = e.apply(n, a), n = a = null);
  }
  const l = function() {
    return n = this, a = arguments, i = Date.now(), s || (s = setTimeout(u, c)), p;
  };
  return l.clear = function() {
    s && (clearTimeout(s), s = null);
  }, l.flush = function() {
    s && (p = e.apply(n, a), n = a = null, clearTimeout(s), s = null);
  }, l;
}
const A = "page-break";
var m = /* @__PURE__ */ ((e) => (e.A5 = "a5", e.A5Portrait = "a5-portrait", e.A4 = "a4", e.A3 = "a3", e.B5 = "b5", e.B4 = "b4", e.letter = "letter", e.legal = "legal", e.ledger = "ledger", e))(m || {});
function _(e, o) {
  const s = {
    mm: 3.7795275590551185,
    // 1mm ≈ 3.78px
    cm: 37.79527559055118,
    // 1cm ≈ 37.8px
    in: 96,
    // 1in = 96px
    pt: 1.3333333333333333,
    // 1pt ≈ 1.33px
    pc: 16
    // 1pc ≈ 16px
  };
  return s[o] ? e * s[o] : null;
}
const ue = (e, o) => {
  const { name: t, id: s, width: a, height: n, unit: i } = e, { fixedHeight: p } = o, c = `${a}${i}`, u = `${n}${i}`;
  return {
    id: s || t.toLowerCase().replace(/\s/g, "-"),
    name: `${t} (${c} x ${u})`,
    sizeH: n,
    sizeW: a,
    sizeUnit: i,
    widthMedia: "",
    width: c,
    minHeight: p ? "0" : `${_(n, i)}px`,
    height: p ? u : void 0
  };
}, H = "page-breaks", S = "gjs-page-breaks-spot", j = (e) => e.type === H;
function he(e) {
  const { Devices: o } = e, { height: t, minHeight: s } = o.getSelected().attributes, n = parseFloat(`${t && t !== "auto" ? t : s}`), i = e.getWrapper(), p = i == null ? void 0 : i.getEl(), c = [];
  if (!i || !p)
    return c;
  const u = p.getBoundingClientRect().height, r = i.findType(A).map((h) => h.getEl()).filter(Boolean).map((h) => ({ top: h.offsetTop, height: h.offsetHeight }));
  let g = 0;
  for (; g < u; ) {
    const h = r.find((y) => y.top > g && y.top < g + n);
    h ? (g = h.top + h.height, c.push(h.top)) : (g += n, g < u && c.push(g));
  }
  return c;
}
function fe(e, o) {
  const { Canvas: t } = e, s = t.events, a = e.events;
  let n;
  const i = () => {
    K(
      `${q}preset-printable-spots`,
      `.${S} {
        position: absolute;
        left: 0;
        width: 100%;
        border-top: 1px dashed var(--gs-theme-cl-cmp-bg2, var(--gjs-color-blue));
        color: var(--gs-theme-cl-prm-txt, var(--gjs-font-color));
      }
      .${S}::after {
        background-color: var(--gs-theme-cl-cmp-bg1, var(--gjs-color-highlight));
        content: attr(data-page-number);
        display: block;
        padding: 0 0.5rem;
        border-radius: 2px;
        position: absolute;
        top: -2px;
        right: 0;
        transform: translate(100%, -50%);
      }
      `,
      { target: window.document.head }
    );
  }, p = () => {
    if (o.enablePageBreaksSpot === !1) return;
    const d = { id: H, type: H, component: e.getWrapper() };
    return t.getSpots(d).pop() || t.addSpot(d);
  }, c = ({ spot: d }) => {
    const { spotEl: f } = d.attributes;
    !j(d) || !f || !n || n.removeChild(f);
  }, u = ({ spot: d }) => {
    if (!j(d) || !n) return;
    let { spotEl: f } = d.attributes;
    f || (f = document.createElement("div"), f.className = `${S}__items`, d.set({ spotEl: f })), n.appendChild(f);
  }, l = ({ spot: d }) => {
    const { spotEl: f, pageBreaks: x = [] } = d.attributes;
    !j(d) || !f || !n || (Object.entries(d.getStyle()).forEach(([L, $]) => f.style.setProperty(L, $)), f.innerHTML = "", x.forEach((L, $) => {
      const E = document.createElement("div");
      E.setAttribute("data-page-number", `${$ + 2}`), E.className = S, E.style.top = `${L}px`, f.appendChild(E);
    }));
  }, r = N(() => {
    if (e.em.destroyed) return;
    const d = p();
    d && d.set({ pageBreaks: he(e), up: (d.attributes.up || 0) + 1 });
  }, 150), g = N(r, 200), h = () => {
    n = t.getSpotsEl(), p();
  }, y = () => {
    const d = p();
    d && d.set({ up: (d.attributes.up || 0) + 1 });
  }, w = [
    ["component:input", r],
    ["frame:scroll", y],
    [a.update, r],
    [s.spotAdd, u],
    [s.spotUpdate, l],
    [s.spotRemove, c],
    [s.frameLoadBody, h]
  ];
  return w.forEach(([d, f]) => e.on(d, f)), e.onReady(() => {
    i(), new ResizeObserver(y).observe(t.getElement().parentElement);
  }), () => {
    w.forEach(([d, f]) => e.off(d, f)), r.clear(), g.clear();
  };
}
var M = /* @__PURE__ */ ((e) => (e.print = "presetPrintable:print", e))(M || {});
const F = (e) => new Promise((o) => {
  const t = document.createElement("iframe");
  t.style.cssText = "position: absolute; width: 0; height: 0; border: 0; opacity: 0; pointer-events: none;", t.srcdoc = e.content, t.onload = () => o(t), document.body.appendChild(t);
});
function ve(e, o) {
  const { Commands: t, Pages: s } = e;
  return t.add(
    M.print,
    async (a, n, i = {}) => {
      const p = {
        page: i.allPages ? void 0 : s.getSelected(),
        styles: "inline",
        optionsHtml: {
          // Avoid DataSource exporter
          exporter: null
        }
      }, u = (await t.run(z.projectFiles, p)).filter((g) => g.filename.endsWith(".html")), l = u.map((g) => g.content).join('<div style="break-before: page;"></div>'), r = [];
      if (i.separatePrints)
        for (const g of u)
          r.push(await F(g));
      else
        r.push(await F({ content: l, filename: "", mimeType: "" }));
      for (const g of r) {
        const h = g.contentWindow;
        h.focus(), h.print();
      }
      setTimeout(() => r.forEach((g) => g.remove()), 1e3);
    }
  ), t.add("presetPrintable:exp", async (a) => {
    await O("https://cdn.jsdelivr.net/npm/html2canvas@1.4.1/dist/html2canvas.min.js"), await O("https://cdn.jsdelivr.net/npm/jspdf@3.0.0/dist/jspdf.umd.min.js");
    const { Devices: n, Canvas: i } = a, { sizeW: p, sizeH: c, sizeUnit: u } = n.getSelected().attributes, l = i.getFrameEl().contentDocument, r = "px", g = [_(p, u), _(c, u)], h = new window.jspdf.jsPDF({ unit: r, format: g }), y = l.body;
    h.html(y, { callback: (w) => w.save() });
  }), () => {
    delete e.Commands.commands[M.print];
  };
}
function be(e, o) {
  const { blockPageBreak: t = {} } = o, { Components: s, Blocks: a } = e, n = "Page Break", i = `${ge}${A}`;
  return s.addType(A, {
    block: t && {
      label: n,
      media: '<svg viewBox="0 0 24 24"><path d="M18 20H6v-2H4v2a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-2h-2v2M14 2H6a2 2 0 0 0-2 2v8h2V4h8v4h4v4h2V8l-6-6m-3 14H8v-2h3v2m5 0h-3v-2h3v2M3 14h3v2H3v-2m18 2h-3v-2h3v2Z"/></svg>',
      category: "Extra",
      select: !0,
      ...t
    },
    isComponent: de(A),
    model: {
      defaults: {
        name: n,
        classes: i,
        attributes: { [V]: A },
        emptyState: !0,
        stylable: !1,
        droppable: !1,
        draggable: pe("wrapper"),
        styles: `.${i} { break-before: page; }`,
        traits: []
      }
    }
  }), () => {
    s.removeType(A), a.remove(A);
  };
}
const me = "presetPrintable", ye = R.free, we = function(e, o = {}) {
  var G;
  const t = {
    selectedDevice: m.A4,
    ...o
  }, { Canvas: s, Devices: a, Blocks: n } = e, i = s.events, p = e.events, c = t.selectedDevice || a.getConfig().default, u = [
    { id: m.A5, name: "A5", width: 148, height: 210, unit: "mm" },
    { id: m.A5Portrait, name: "A5 Portrait", width: 210, height: 148, unit: "mm" },
    { id: m.A4, name: "A4", width: 210, height: 297, unit: "mm" },
    { id: m.A3, name: "A3", width: 297, height: 420, unit: "mm" },
    { id: m.B5, name: "B5", width: 176, height: 250, unit: "mm" },
    { id: m.B4, name: "B4", width: 250, height: 353, unit: "mm" },
    { id: m.letter, name: "Letter", width: 8.5, height: 11, unit: "in" },
    { id: m.legal, name: "Legal", width: 8.5, height: 14, unit: "in" },
    { id: m.ledger, name: "Ledger", width: 11, height: 17, unit: "in" }
  ], r = (((G = t.devices) == null ? void 0 : G.call(t, { items: u })) || u).map((v) => ue(v, t));
  a.devices.reset(r);
  const g = c && a.get(c) || a.getAll().at(0);
  a.select(g), ["map", "video"].forEach((v) => n.remove(v));
  const y = (v = {}) => {
    const b = "page-rule", { rules: P } = e.Css, { width: T, height: B, minHeight: k } = a.getSelected().attributes, Y = B && B !== "auto" ? B : k, D = P.where({ id: b }).pop(), J = D || P.add(
      {
        id: b,
        selectors: [],
        selectorsAdd: "@page"
      },
      { at: 0 }
    );
    (v.fromDeviceChange || !D) && J.addStyle({
      size: `${T} ${Y}`,
      ...!D && { margin: "0" }
    });
  }, w = (v, b = {}) => {
    const P = v.document, { minHeight: T } = a.getSelected().attributes, k = `
      ${!!parseFloat(`${T}`) ? `
        [data-gjs-type="wrapper"] { min-height: ${T}; }
      ` : `
        body {
          overflow: hidden;
        }

        [data-gjs-type="wrapper"] {
            overflow: hidden;
            height: 100dvh;
        }
      `}
    `;
    K(`${q}preset-printable`, k, { doc: P, target: P.head }), y(b);
  };
  e.on(i.frameLoadBody, (v) => {
    const b = v.window;
    if (w(b), t.fixedHeight) {
      const T = v.model.getComponent().getEl();
      T == null || T.addEventListener("scroll", (B) => {
        const k = B.target;
        k.scrollTop > 0 && k.scrollTo(0, 0);
      });
    }
  }), e.on(a.events.select, () => {
    var b;
    const v = (b = s.getFrameEl().contentDocument) == null ? void 0 : b.defaultView;
    v && w(v, { fromDeviceChange: !0 });
  });
  const d = () => {
    const b = e.Css.getRules().filter((P) => P.selectorsToString() === "body").pop();
    b == null || b.addStyle({
      "margin-top": "",
      "margin-right": "",
      "margin-bottom": "",
      "margin-left": "",
      margin: "0"
    });
  }, f = [[p.projectLoad, d]];
  f.forEach(([v, b]) => e.on(v, b));
  const x = fe(e, t), L = ve(e), $ = be(e, t);
  e.onReady(() => {
    e.UndoManager.clear();
  });
  const E = () => {
    f.forEach(([v, b]) => e.off(v, b)), x(), L(), $();
  };
  e.on(p.destroy, E), ce({
    editor: e,
    licenseKey: t.licenseKey,
    plan: ye,
    pluginName: me,
    cleanup: E
  });
}, Pe = re(we);
export {
  Pe as default
};
