// @storefront-ui/vue/dist/components/SfDrawer/SfDrawer.vue.mjs
import { defineComponent as p, toRefs as w, ref as y, computed as g, openBlock as C, createBlock as b, resolveDynamicComponent as v, mergeProps as h, unref as r, withKeys as k, withCtx as V, renderSlot as B, createCommentVNode as S } from "vue";
import { onClickOutside as D } from "@vueuse/core";
import { useTwMergeRoot as R, twMerge as x } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfDrawerPlacement as e } from "@storefront-ui/shared";
const P = /* @__PURE__ */ p({
  inheritAttrs: !1,
  __name: "SfDrawer",
  props: {
    modelValue: {
      type: Boolean,
      default: !1
    },
    placement: {
      type: String,
      default: e.left
    },
    tag: {
      type: [String, Object],
      default: "aside"
    },
    disableClickAway: {
      type: Boolean,
      default: !1
    },
    disableEsc: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update:modelValue"],
  setup(a, { emit: n }) {
    const f = a, { attrsWithoutClass: d } = R(), { disableClickAway: u, disableEsc: i, placement: t } = w(f), l = n, o = y();
    D(o, () => {
      u.value || l("update:modelValue", !1);
    });
    const c = () => {
      i.value || l("update:modelValue", !1);
    }, m = g(() => ({
      "left-0": t.value !== e.right,
      "right-0": t.value !== e.left,
      "top-0": t.value !== e.bottom,
      "bottom-0": t.value !== e.top
    }));
    return (s, A) => a.modelValue ? (C(), b(v(a.tag), h({
      key: 0,
      ref_key: "drawerRef",
      ref: o,
      class: r(x)("fixed", m.value, s.$attrs.class),
      "data-testid": "drawer",
      tabindex: "-1"
    }, r(d), {
      onKeydown: k(c, ["esc"])
    }), {
      default: V(() => [
        B(s.$slots, "default")
      ]),
      _: 3
    }, 16, ["class"])) : S("", !0);
  }
});
export {
  P as default
};
