// @polar-sh/sdk/dist/esm/models/components/benefitmetercreditcreate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { BenefitMeterCreditCreateProperties$outboundSchema, } from "./benefitmetercreditcreateproperties.js";
/** @internal */
export const BenefitMeterCreditCreateMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitMeterCreditCreateMetadataToJSON(benefitMeterCreditCreateMetadata) {
    return JSON.stringify(BenefitMeterCreditCreateMetadata$outboundSchema.parse(benefitMeterCreditCreateMetadata));
}
/** @internal */
export const BenefitMeterCreditCreate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("meter_credit"),
    description: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: BenefitMeterCreditCreateProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function benefitMeterCreditCreateToJSON(benefitMeterCreditCreate) {
    return JSON.stringify(BenefitMeterCreditCreate$outboundSchema.parse(benefitMeterCreditCreate));
}
