// @polar-sh/sdk/dist/esm/models/components/benefitlicensekeysupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { BenefitLicenseKeysCreateProperties$outboundSchema, } from "./benefitlicensekeyscreateproperties.js";
/** @internal */
export const BenefitLicenseKeysUpdateMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitLicenseKeysUpdateMetadataToJSON(benefitLicenseKeysUpdateMetadata) {
    return JSON.stringify(BenefitLicenseKeysUpdateMetadata$outboundSchema.parse(benefitLicenseKeysUpdateMetadata));
}
/** @internal */
export const BenefitLicenseKeysUpdate$outboundSchema = z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    description: z.optional(z.nullable(z.string())),
    type: z.literal("license_keys"),
    properties: z.optional(z.nullable(BenefitLicenseKeysCreateProperties$outboundSchema)),
});
export function benefitLicenseKeysUpdateToJSON(benefitLicenseKeysUpdate) {
    return JSON.stringify(BenefitLicenseKeysUpdate$outboundSchema.parse(benefitLicenseKeysUpdate));
}
