// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablesupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { BenefitDownloadablesCreateProperties$outboundSchema, } from "./benefitdownloadablescreateproperties.js";
/** @internal */
export const BenefitDownloadablesUpdateMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitDownloadablesUpdateMetadataToJSON(benefitDownloadablesUpdateMetadata) {
    return JSON.stringify(BenefitDownloadablesUpdateMetadata$outboundSchema.parse(benefitDownloadablesUpdateMetadata));
}
/** @internal */
export const BenefitDownloadablesUpdate$outboundSchema = z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    description: z.optional(z.nullable(z.string())),
    type: z.literal("downloadables"),
    properties: z.optional(z.nullable(BenefitDownloadablesCreateProperties$outboundSchema)),
});
export function benefitDownloadablesUpdateToJSON(benefitDownloadablesUpdate) {
    return JSON.stringify(BenefitDownloadablesUpdate$outboundSchema.parse(benefitDownloadablesUpdate));
}
