// @polar-sh/sdk/dist/esm/models/components/customerupdateexternalid.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { AddressInput$outboundSchema, } from "./addressinput.js";
import { TaxIDFormat$outboundSchema } from "./taxidformat.js";
/** @internal */
export const CustomerUpdateExternalIDMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function customerUpdateExternalIDMetadataToJSON(customerUpdateExternalIDMetadata) {
    return JSON.stringify(CustomerUpdateExternalIDMetadata$outboundSchema.parse(customerUpdateExternalIDMetadata));
}
/** @internal */
export const CustomerUpdateExternalIDTaxID$outboundSchema = z.union([z.string(), TaxIDFormat$outboundSchema]);
export function customerUpdateExternalIDTaxIDToJSON(customerUpdateExternalIDTaxID) {
    return JSON.stringify(CustomerUpdateExternalIDTaxID$outboundSchema.parse(customerUpdateExternalIDTaxID));
}
/** @internal */
export const CustomerUpdateExternalID$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    email: z.optional(z.nullable(z.string())),
    name: z.optional(z.nullable(z.string())),
    billingAddress: z.optional(z.nullable(AddressInput$outboundSchema)),
    taxId: z.optional(z.nullable(z.array(z.nullable(z.union([z.string(), TaxIDFormat$outboundSchema]))))),
}), z.transform((v) => {
    return remap$(v, {
        billingAddress: "billing_address",
        taxId: "tax_id",
    });
}));
export function customerUpdateExternalIDToJSON(customerUpdateExternalID) {
    return JSON.stringify(CustomerUpdateExternalID$outboundSchema.parse(customerUpdateExternalID));
}
