// @polar-sh/sdk/dist/esm/models/components/customerupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { AddressInput$outboundSchema, } from "./addressinput.js";
import { TaxIDFormat$outboundSchema } from "./taxidformat.js";
/** @internal */
export const CustomerUpdateMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function customerUpdateMetadataToJSON(customerUpdateMetadata) {
    return JSON.stringify(CustomerUpdateMetadata$outboundSchema.parse(customerUpdateMetadata));
}
/** @internal */
export const CustomerUpdateTaxId$outboundSchema = z.union([z.string(), TaxIDFormat$outboundSchema]);
export function customerUpdateTaxIdToJSON(customerUpdateTaxId) {
    return JSON.stringify(CustomerUpdateTaxId$outboundSchema.parse(customerUpdateTaxId));
}
/** @internal */
export const CustomerUpdate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    email: z.optional(z.nullable(z.string())),
    name: z.optional(z.nullable(z.string())),
    billingAddress: z.optional(z.nullable(AddressInput$outboundSchema)),
    taxId: z.optional(z.nullable(z.array(z.nullable(z.union([z.string(), TaxIDFormat$outboundSchema]))))),
    externalId: z.optional(z.nullable(z.string())),
}), z.transform((v) => {
    return remap$(v, {
        billingAddress: "billing_address",
        taxId: "tax_id",
        externalId: "external_id",
    });
}));
export function customerUpdateToJSON(customerUpdate) {
    return JSON.stringify(CustomerUpdate$outboundSchema.parse(customerUpdate));
}
