// @polar-sh/sdk/dist/esm/models/components/customfieldcreatenumber.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { CustomFieldNumberProperties$outboundSchema, } from "./customfieldnumberproperties.js";
/** @internal */
export const CustomFieldCreateNumberMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function customFieldCreateNumberMetadataToJSON(customFieldCreateNumberMetadata) {
    return JSON.stringify(CustomFieldCreateNumberMetadata$outboundSchema.parse(customFieldCreateNumberMetadata));
}
/** @internal */
export const CustomFieldCreateNumber$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("number"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: CustomFieldNumberProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function customFieldCreateNumberToJSON(customFieldCreateNumber) {
    return JSON.stringify(CustomFieldCreateNumber$outboundSchema.parse(customFieldCreateNumber));
}
