// @polar-sh/sdk/dist/esm/models/components/customfieldcreatedate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { CustomFieldDateProperties$outboundSchema, } from "./customfielddateproperties.js";
/** @internal */
export const CustomFieldCreateDateMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function customFieldCreateDateMetadataToJSON(customFieldCreateDateMetadata) {
    return JSON.stringify(CustomFieldCreateDateMetadata$outboundSchema.parse(customFieldCreateDateMetadata));
}
/** @internal */
export const CustomFieldCreateDate$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    type: z.literal("date"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.optional(z.nullable(z.string())),
    properties: CustomFieldDateProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        organizationId: "organization_id",
    });
}));
export function customFieldCreateDateToJSON(customFieldCreateDate) {
    return JSON.stringify(CustomFieldCreateDate$outboundSchema.parse(customFieldCreateDate));
}
