// @storefront-ui/vue/dist/components/SfCounter/SfCounter.vue.mjs
import { defineComponent as c, toRefs as p, computed as f, openBlock as i, createElementBlock as x, mergeProps as m, unref as n, renderSlot as d } from "vue";
import { useTwMergeRoot as g, twMerge as v } from "../../composables/useTwMergeRoot/useTwMergeRoot.mjs";
import { SfCounterSize as t } from "@storefront-ui/shared";
const _ = /* @__PURE__ */ c({
  inheritAttrs: !1,
  __name: "SfCounter",
  props: {
    size: {
      type: String,
      default: t.base
    },
    pill: {
      type: Boolean,
      default: !1
    }
  },
  setup(a) {
    const o = a, { attrsWithoutClass: l } = g(), { size: r, pill: e } = p(o), u = f(() => {
      switch (r == null ? void 0 : r.value) {
        case t["3xs"]:
          return ["text-3xs", { "px-1": e.value }];
        case t["2xs"]:
          return ["text-2xs", { "px-1.5": e.value }];
        case t.xs:
          return ["text-xs", { "px-2": e.value }];
        case t.sm:
          return ["text-sm", { "px-2.5": e.value }];
        case t.lg:
          return ["text-lg", { "px-3.5": e.value }];
        default:
          return ["text-base", { "px-3": e.value }];
      }
    });
    return (s, C) => (i(), x("span", m({
      class: n(v)(
        "inline-flex items-center before:content-['('] after:content-[')']",
        ...u.value,
        {
          "rounded-full py-0.5 font-medium ring-1 ring-neutral-200 ring-inset text-neutral-900 before:content-none after:content-none": n(e),
          "text-neutral-500": !n(e)
        },
        s.$attrs.class
      ),
      "data-testid": "counter"
    }, n(l)), [
      d(s.$slots, "default")
    ], 16));
  }
});
export {
  _ as default
};
