// @polar-sh/sdk/dist/commonjs/models/components/seatclaiminfo.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeatClaimInfo$ = exports.SeatClaimInfo$outboundSchema = exports.SeatClaimInfo$inboundSchema = void 0;
exports.seatClaimInfoToJSON = seatClaimInfoToJSON;
exports.seatClaimInfoFromJSON = seatClaimInfoFromJSON;
const z = __importStar(require("zod/v3"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.SeatClaimInfo$inboundSchema = z.object({
    product_name: z.string(),
    product_id: z.string(),
    organization_name: z.string(),
    organization_slug: z.string(),
    customer_email: z.string(),
    can_claim: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "product_name": "productName",
        "product_id": "productId",
        "organization_name": "organizationName",
        "organization_slug": "organizationSlug",
        "customer_email": "customerEmail",
        "can_claim": "canClaim",
    });
});
/** @internal */
exports.SeatClaimInfo$outboundSchema = z.object({
    productName: z.string(),
    productId: z.string(),
    organizationName: z.string(),
    organizationSlug: z.string(),
    customerEmail: z.string(),
    canClaim: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        productName: "product_name",
        productId: "product_id",
        organizationName: "organization_name",
        organizationSlug: "organization_slug",
        customerEmail: "customer_email",
        canClaim: "can_claim",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SeatClaimInfo$;
(function (SeatClaimInfo$) {
    /** @deprecated use `SeatClaimInfo$inboundSchema` instead. */
    SeatClaimInfo$.inboundSchema = exports.SeatClaimInfo$inboundSchema;
    /** @deprecated use `SeatClaimInfo$outboundSchema` instead. */
    SeatClaimInfo$.outboundSchema = exports.SeatClaimInfo$outboundSchema;
})(SeatClaimInfo$ || (exports.SeatClaimInfo$ = SeatClaimInfo$ = {}));
function seatClaimInfoToJSON(seatClaimInfo) {
    return JSON.stringify(exports.SeatClaimInfo$outboundSchema.parse(seatClaimInfo));
}
function seatClaimInfoFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SeatClaimInfo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SeatClaimInfo' from JSON`);
}
